From ced1e0e6e0e7b80ad4e9b9766e0a132dc97d6bb6 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Tue, 14 Feb 2012 20:54:31 +0100
Subject: [PATCH 1/8] depend on eina not eina-0

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 configure.ac |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index c0a5697..0b67de8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -15,7 +15,7 @@ AC_HEADER_STDC
 
 AM_PROG_LIBTOOL
 
-PKG_CHECK_MODULES(TASKS, [elementary eina-0 dbus-1 sqlite3])
+PKG_CHECK_MODULES(TASKS, [elementary eina dbus-1 sqlite3])
 
 AC_OUTPUT([
 Makefile
-- 
1.7.8.4

