From 4ca9e2aac3e9f1a981b48552c3aff255863cd1fe Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Sat, 17 Dec 2011 03:12:29 +0100
Subject: [PATCH 4/8] fix changed callback name

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/gui.c  |    2 +-
 src/main.c |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gui.c b/src/gui.c
index f6f1c6c..69b06d8 100644
--- a/src/gui.c
+++ b/src/gui.c
@@ -1230,7 +1230,7 @@ void edit_cat(void)
 
 	cat_win = elm_win_add(NULL, "cat", ELM_WIN_BASIC);
 	elm_win_title_set(cat_win, "Edit Categories");
-	evas_object_smart_callback_add(cat_win, "delete-request", cat_win_del, cat_win);
+	evas_object_smart_callback_add(cat_win, "delete,request", cat_win_del, cat_win);
 	
 	//add background
 	bg = elm_bg_add(cat_win);
diff --git a/src/main.c b/src/main.c
index 12072f5..116b84c 100644
--- a/src/main.c
+++ b/src/main.c
@@ -64,7 +64,7 @@ elm_main(int argc, char **argv)
 	//set up win
 	win = elm_win_add(NULL, "tasks", ELM_WIN_BASIC);
 	elm_win_title_set(win, "Tasks");
-	evas_object_smart_callback_add(win, "delete-request", my_win_del, NULL);
+	evas_object_smart_callback_add(win, "delete,request", my_win_del, NULL);
 
 	//open database 
 	open_database();
-- 
1.7.8.4

