From d1494e2e59d99cd2b7948a870dca00ae789431b9 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Tue, 14 Feb 2012 20:52:05 +0100
Subject: [PATCH 6/8] remove deprecated API

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/gui.c |   64 ++++++++++++++++++++++++++++++------------------------------
 1 files changed, 32 insertions(+), 32 deletions(-)

diff --git a/src/gui.c b/src/gui.c
index 0291378..c798f75 100644
--- a/src/gui.c
+++ b/src/gui.c
@@ -75,7 +75,7 @@ void select_category(void *data, Evas_Object *obj, void *event_info)
 	if (last_rec != -1) { //cater for initial null data
 		sprintf(te_data, "%s", (const char *)elm_entry_entry_get(tk));
 		task_entry = strtok(te_data, "<");
-		tsk = (_Task *)elm_genlist_item_data_get(task_list[last_rec]);
+		tsk = (_Task *)elm_object_item_data_get(task_list[last_rec]);
 		if (strcmp(task_entry, tsk->text) != 0) {
 			sprintf(tsk->text, "%s", task_entry);
 			WRITE = 1;
@@ -90,7 +90,7 @@ void select_category(void *data, Evas_Object *obj, void *event_info)
 	
 	Evas_Object *li = data;
 	item = (Elm_Object_Item *)elm_genlist_selected_item_get(li);
-	tsk = (_Task *)elm_genlist_item_data_get(item);
+	tsk = (_Task *)elm_object_item_data_get(item);
 	last_rec = tsk->no;
 	elm_object_text_set(cat_bt, tsk->cat);
 	sprintf(tystr, "%d", tsk->pr);
@@ -117,7 +117,7 @@ void set_priority(void *data, Evas_Object *obj, void *event_info)
 		dummy_pr = atoi(prio);
 		return;
 	}
-	_Task *tsk = (_Task *)elm_genlist_item_data_get(item);
+	_Task *tsk = (_Task *)elm_object_item_data_get(item);
 	if(tsk->pr == ty) return;
 	tsk->pr = ty;
 	WRITE = 1;
@@ -131,13 +131,13 @@ void set_category(void *data, Evas_Object *obj, void *event_info)
 	item = (Elm_Object_Item *)elm_genlist_selected_item_get(list);
 	evas_object_hide(hs);
 	if (!item) return;
-	_Task *tsk = (_Task *)elm_genlist_item_data_get(item);
+	_Task *tsk = (_Task *)elm_object_item_data_get(item);
 	if (strcmp(tsk->cat, category) == 0) return;
 	strcpy(tsk->cat, category);
 	elm_object_text_set(cat_bt, category);
 	WRITE = 1;
     if (strcmp(sel_category, " All Tasks ") !=0 &&
-        strcmp(sel_category, category) != 0) elm_genlist_item_del(item);
+        strcmp(sel_category, category) != 0) elm_object_item_del(item);
 }
 
 void set_date(void *data, Evas_Object *obj, void *event_info)
@@ -151,7 +151,7 @@ void set_date(void *data, Evas_Object *obj, void *event_info)
 		strcpy(dummy_date, dt);
 		return;
 	}
-	_Task *tsk = (_Task *)elm_genlist_item_data_get(item);
+	_Task *tsk = (_Task *)elm_object_item_data_get(item);
 	if (strcmp(tsk->date, dt) == 0) return;
 	strcpy(tsk->date, dt);
 	WRITE = 1;
@@ -167,7 +167,7 @@ void task_cb_changed(void *data, Evas_Object *obj, void *event_info)
 	//printf("no %d key %d\n", tsk->no, tsk->key);
 	//TODO - strike through the task
 	//remove this task from the list
-	elm_genlist_item_del(task_list[tsk->no]);
+	elm_object_item_del(task_list[tsk->no]);
 	//update the db
 	update_record (tsk->no);
 	last_rec = -1;
@@ -218,7 +218,7 @@ void set_create_date(void *data, Evas_Object *obj, void *event_info)
 	Details *det = data;
 	evas_object_hide(det->hover);
 	//change date value of selected task
-	_Task *tsk = (_Task *)elm_genlist_item_data_get(det->item);
+	_Task *tsk = (_Task *)elm_object_item_data_get(det->item);
 	strcpy(tsk->date, det->data);
 	//update button label
 	elm_object_text_set(det->button, det->data);
@@ -230,7 +230,7 @@ void set_create_priority(void *data, Evas_Object *obj, void *event_info)
 	Details *det = data;
 	evas_object_hide(det->hover);
 	//change priority value of selected task
-	_Task *tsk = (_Task *)elm_genlist_item_data_get(det->item);
+	_Task *tsk = (_Task *)elm_object_item_data_get(det->item);
 	tsk->pr = atoi(det->data);
 	//update button label
 	elm_object_text_set(det->button, det->data);
@@ -295,7 +295,7 @@ void create_details_page(void *data, Evas_Object *obj, void *event_info)
 	//check task selected - otherwise return
 	Elm_Object_Item *item = (Elm_Object_Item *)elm_genlist_selected_item_get(list);
 	if(!item) return;
-	_Task *tsk = (_Task *)elm_genlist_item_data_get(item);
+	_Task *tsk = (_Task *)elm_object_item_data_get(item);
 	det_page = elm_win_add(NULL, "inwin", ELM_WIN_BASIC);
 	elm_win_title_set(det_page, "Task Details");
 	elm_win_autodel_set(det_page, 1);
@@ -322,7 +322,7 @@ void create_details_page(void *data, Evas_Object *obj, void *event_info)
 	lb = elm_label_add(det_page);
 	elm_object_text_set(lb,"Edit Task Details");
 	evas_object_size_hint_align_set(lb, 0.5, -1.0);
-	elm_frame_content_set(fr, lb);
+	elm_object_content_set(fr, lb);
 	evas_object_show(lb);
 
 	//add a label 
@@ -366,7 +366,7 @@ void create_details_page(void *data, Evas_Object *obj, void *event_info)
 
 	bt = elm_button_add(det_page);
 	elm_object_text_set(bt, "Pr");
-	elm_hover_content_set(det_hv, "middle", bt);
+	elm_object_part_content_set(det_hv, "middle", bt);
 	evas_object_show(bt);
 
 	bx = elm_box_add(det_page);
@@ -384,7 +384,7 @@ void create_details_page(void *data, Evas_Object *obj, void *event_info)
 		evas_object_show(bt);
 	}	
 	evas_object_show(bx);
-	elm_hover_content_set(det_hv, "bottom", bx);
+	elm_object_part_content_set(det_hv, "bottom", bx);
 
 	//add a label to hbox
 	lb = elm_label_add(det_page);
@@ -409,14 +409,14 @@ void create_details_page(void *data, Evas_Object *obj, void *event_info)
 
 	bt = elm_button_add(det_page);
 	elm_object_text_set(bt, "Category");
-	elm_hover_content_set(hs, "middle", bt);
+	elm_object_part_content_set(hs, "middle", bt);
 	evas_object_show(bt);
 
 	bx = elm_box_add(det_page);
 	//add categories
 	add_hs_items (det_page, bx, bt, 0);
 	evas_object_show(bx);
-	elm_hover_content_set(hs, "bottom", bx); 
+	elm_object_part_content_set(hs, "bottom", bx); 
 
 	//add a label to hbox
 	lb = elm_label_add(det_page);
@@ -442,7 +442,7 @@ void create_details_page(void *data, Evas_Object *obj, void *event_info)
 
 	bt = elm_button_add(det_page);
 	elm_object_text_set(bt, "Date");
-	elm_hover_content_set(c_date_hs, "middle", bt);
+	elm_object_part_content_set(c_date_hs, "middle", bt);
 	evas_object_show(bt);
 
 	bx = elm_box_add(det_page);
@@ -486,7 +486,7 @@ void create_details_page(void *data, Evas_Object *obj, void *event_info)
 	evas_object_show(bt);
 	
 	evas_object_show(bx);
-	elm_hover_content_set(c_date_hs, "bottom", bx);
+	elm_object_part_content_set(c_date_hs, "bottom", bx);
 
 	//add a label 
 	lb = elm_label_add(det_page);
@@ -568,7 +568,7 @@ void save_button_clicked(void *data, Evas_Object *obj, void *event_info)
 	//get task no
 	Elm_Object_Item *item = (Elm_Object_Item *)elm_genlist_selected_item_get(list);
 	if (item) {
-		_Task *tsk = (_Task *)elm_genlist_item_data_get(item);
+		_Task *tsk = (_Task *)elm_object_item_data_get(item);
 		
 		//get data from entry
 		sprintf(te_data, "%s", (const char *)elm_entry_entry_get(tk));
@@ -627,7 +627,7 @@ void note_done(void *data, Evas_Object *obj, void *event_info)
 	//Evas_Object *en = data;
 	
 	Elm_Object_Item *item = (Elm_Object_Item *)elm_genlist_selected_item_get(list);
-	_Task *tsk = (_Task *)elm_genlist_item_data_get(item);
+	_Task *tsk = (_Task *)elm_object_item_data_get(item);
 	sprintf(tystr, "%s", elm_entry_entry_get(data));
 	if (strcmp(tystr, "<br>") == 0) strcpy(tystr, "");
 	strcpy(tsk->note, tystr);
@@ -643,7 +643,7 @@ void note_button_clicked(void *data, Evas_Object *obj, void *event_info)
 
 	Elm_Object_Item *item = (Elm_Object_Item *)elm_genlist_selected_item_get(list);
 	if(!item) return;
-	_Task *tsk = (_Task *)elm_genlist_item_data_get(item);
+	_Task *tsk = (_Task *)elm_object_item_data_get(item);
 
 	note_win = elm_win_add(NULL, "note", ELM_WIN_BASIC);
 	elm_win_title_set(note_win, "Note");
@@ -721,7 +721,7 @@ void create_cat_hover(void)
 	cat_hv_bx = elm_box_add(win);
 	add_hs_items (win, cat_hv_bx, bt, 1);
 	evas_object_show(cat_hv_bx);
-	elm_hover_content_set(hs1, "top", cat_hv_bx);
+	elm_object_part_content_set(hs1, "top", cat_hv_bx);
 	evas_object_hide(hs1);
 }
 
@@ -880,7 +880,7 @@ void create_gui(Evas_Object *win)
 
 	bt = elm_button_add(win);
 	elm_object_text_set(bt, "Pr");
-	elm_hover_content_set(hv, "middle", bt);
+	elm_object_part_content_set(hv, "middle", bt);
 	evas_object_show(bt);
 
 	bx = elm_box_add(win);
@@ -895,7 +895,7 @@ void create_gui(Evas_Object *win)
 		evas_object_show(bt);
 	}	
 	evas_object_show(bx);
-	elm_hover_content_set(hv, "bottom", bx);
+	elm_object_part_content_set(hv, "bottom", bx);
 	
 	//add scroller for entry
 	sc = elm_scroller_add(win);
@@ -911,7 +911,7 @@ void create_gui(Evas_Object *win)
 	elm_entry_entry_set(tk, "Task");
 	evas_object_size_hint_weight_set(tk, 1.0, 0.0);
 	evas_object_size_hint_align_set(tk, -1.0, 0.0);
-	elm_scroller_content_set(sc, tk);
+	elm_object_content_set(sc, tk);
 	evas_object_show(tk);
 	evas_object_show(sc);
 
@@ -934,13 +934,13 @@ void create_gui(Evas_Object *win)
 
 	bt = elm_button_add(win);
 	elm_object_text_set(bt, "Date");
-	elm_hover_content_set(date_hs, "middle", bt);
+	elm_object_part_content_set(date_hs, "middle", bt);
 	evas_object_show(bt);
 
 	bx1 = elm_box_add(win);
 	add_dates(win, bx1, bt);
 	evas_object_show(bx1);
-	elm_hover_content_set(date_hs, "bottom", bx1);
+	elm_object_part_content_set(date_hs, "bottom", bx1);
 
 	//add save button
 	save_button = elm_button_add(win);
@@ -957,7 +957,7 @@ void create_gui(Evas_Object *win)
 	evas_object_size_hint_align_set(list, -1.0, -1.0);
 	elm_list_multi_select_set(list, 0);
 	elm_box_pack_end(vbox, list);
-	elm_genlist_horizontal_mode_set(list, ELM_LIST_LIMIT);
+	elm_genlist_mode_set(list, ELM_LIST_LIMIT);
 	evas_object_show(list);
 	evas_object_smart_callback_add(list, "selected", task_select, list);
 
@@ -1027,7 +1027,7 @@ void create_gui(Evas_Object *win)
 
 	bt = elm_button_add(win);
 	elm_object_text_set(bt, " All Tasks ");
-	elm_hover_content_set(hs1, "middle", bt);
+	elm_object_part_content_set(hs1, "middle", bt);
 	evas_object_show(bt);
 	create_cat_hover ();
 
@@ -1118,7 +1118,7 @@ void create_cat_dialog(void *data, Evas_Object *obj, void *event_info)
 	vbox = elm_box_add(cat_dialog);
 	elm_win_resize_object_add(cat_dialog, vbox);
 	evas_object_size_hint_weight_set(vbox, 1.0, 1.0);
-	elm_win_inwin_content_set(inwin, vbox);
+	elm_object_content_set(inwin, vbox);
 	evas_object_show(vbox);
 
 	//add hbox to vbox
@@ -1194,11 +1194,11 @@ void  del_cat_button_clicked(void *data, Evas_Object *obj, void *event_info)
 {
 	Evas_Object *li = data;
 	Elm_Object_Item *item = (Elm_Object_Item *)elm_genlist_selected_item_get(li);
-	char *_cat = (char *)elm_genlist_item_data_get(item);
+	char *_cat = (char *)elm_object_item_data_get(item);
 	//delete category from db
 	del_category(_cat);
 	//remove item from list
-	elm_genlist_item_del(item);
+	elm_object_item_del(item);
 	create_cat_hover ();
 	if(strcmp(sel_category, _cat) == 0) select_category (" All Tasks ", NULL, NULL);
 }
@@ -1263,7 +1263,7 @@ void edit_cat(void)
 	// add a label
 	lb = elm_label_add(cat_win);
 	elm_object_text_set(lb, "Task Categories");
-	elm_frame_content_set(fr, lb);
+	elm_object_content_set(fr, lb);
 	evas_object_show(lb);  
 	
 	//add list to vbox now
-- 
1.7.8.4

