From 8f5979f9c289b4c9313dd93a113ed164d3acb5a2 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Tue, 14 Feb 2012 21:08:39 +0100
Subject: [PATCH 7/8] db_sqlite: fix warning

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/db_sqlite.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/db_sqlite.c b/src/db_sqlite.c
index 3eb7790..326063e 100644
--- a/src/db_sqlite.c
+++ b/src/db_sqlite.c
@@ -19,7 +19,7 @@
 
 void open_database(void)
 {
-	int db_ret, ret;
+	int db_ret;
 	char db[255];
 	
 	sprintf(db, "%s/.tasks/tasks.db", home_dir);
@@ -28,7 +28,7 @@ void open_database(void)
 		printf("SQL error: %s\n", sqlite3_errmsg(tasks));
 		printf("Creating new db file\n");
 		//exit(1);
-		ret = system("mkdir ~/.tasks");
+		system("mkdir ~/.tasks");
 		sqlite3_close(tasks);
 		db_ret = sqlite3_open(db, &tasks);
 	}
-- 
1.7.8.4

