From aa85ab2be6be7cb3c4df68139a8271fc4c8c11c8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Picca=20Fr=C3=A9d=C3=A9ric-Emmanuel?= <picca@synchrotron-soleil.fr>
Date: Tue, 1 Mar 2011 21:49:23 +0100
Subject: [PATCH] * fix the QAssitant ftbfs

---
 configure.py |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.py b/configure.py
index 7ed9582..185ec9f 100644
--- a/configure.py
+++ b/configure.py
@@ -354,10 +354,10 @@ class ConfigurePyQt4:
 
         if opts.mwg_ssl_dir:
             ass_lib_dirs = [os.path.join(opts.mwg_ssl_dir, "lib")]
-            ass_libs = ["ssleay32", "libeay32"]
+            ass_libs = ["ssleay32", "libeay32", "QtCore"]
         else:
             ass_lib_dirs = None
-            ass_libs = None
+            ass_libs = ["QtCore"]
 
         # Note that the order in which we check is important for the
         # consolidated module - a module's dependencies must be checked first.
-- 
1.7.4.1

