Description: Fix QtHelp FTBFS with ld --no-add-needed.
Author: Felix Geyer <debfx-pkg@fobos.de>

--- python-qt4-4.8.1.orig/configure.py
+++ python-qt4-4.8.1/configure.py
@@ -364,7 +364,8 @@ class ConfigurePyQt4:
         pyqt_modules.append("QtCore")
 
         check_module("QtGui", "qwidget.h", "new QWidget()")
-        check_module("QtHelp", "qhelpengine.h", "new QHelpEngine(\"foo\")")
+        check_module("QtHelp", "qhelpengine.h", "new QHelpEngine(\"foo\")",
+                extra_libs=["QtCore"])
         check_module("QtMultimedia", "QAudioDeviceInfo",
                 "new QAudioDeviceInfo()")
         check_module("QtNetwork", "qhostaddress.h", "new QHostAddress()")
