From 3237f335c5b559c68749173f4343e6475f9eb8c7 Mon Sep 17 00:00:00 2001
From: Florent Revest <revestflo@gmail.com>
Date: Mon, 30 Oct 2017 21:05:46 +0100
Subject: [PATCH] Allow adbd to be ran as root

---
 adb/adb.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/adb/adb.c b/adb/adb.c
index 10a1e0d..a88641f 100644
--- a/adb/adb.c
+++ b/adb/adb.c
@@ -1267,6 +1267,7 @@ static int should_drop_privileges() {
     int secure = 0;
     char value[PROPERTY_VALUE_MAX];
 
+    return 0;
    /* run adbd in secure mode if ro.secure is set and
     ** we are not in the emulator
     */
-- 
2.14.3

