Description: fix for upstream issue 747, clang 6 build error
Forwarded: https://github.com/Microsoft/cpprestsdk/pull/844
From: rozhuk-im
--- a/Release/include/pplx/pplxlinux.h	2018-04-29 16:22:39.927675000 +0300
+++ b/Release/include/pplx/pplxlinux.h	2018-04-29 16:22:57.809537000 +0300
@@ -240,6 +240,7 @@
     {
     public:
         _PPLXIMP virtual void schedule( TaskProc_t proc, _In_ void* param);
+    virtual ~linux_scheduler() {}
     };
 
 } // namespace details
