Index: SDL_ttf-2.0.10/configure.in
===================================================================
--- SDL_ttf-2.0.10.orig/configure.in	2009-10-12 23:06:38.000000000 +0000
+++ SDL_ttf-2.0.10/configure.in	2014-07-17 12:33:34.011662505 +0000
@@ -1,5 +1,4 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(README)
 
 dnl Set various version strings - taken gratefully from the GTk sources
 
@@ -11,12 +10,19 @@
 # if backwards compatibility has been broken,
 # set BINARY_AGE and INTERFACE_AGE to 0.
 
-MAJOR_VERSION=2
-MINOR_VERSION=0
-MICRO_VERSION=10
+m4_define([sdlttf_major_version],[2])
+m4_define([sdlttf_minor_version],[0])
+m4_define([sdlttf_micro_version],[10])
+m4_define([sdlttf_version], [sdlttf_major_version.sdlttf_minor_version.sdlttf_micro_version])
+
+AC_INIT([SDL_ttf], [sdlttf_version])
+
+MAJOR_VERSION=sdlttf_major_version
+MINOR_VERSION=sdlttf_minor_version
+MICRO_VERSION=sdlttf_micro_version
 INTERFACE_AGE=0
 BINARY_AGE=10
-VERSION=$MAJOR_VERSION.$MINOR_VERSION.$MICRO_VERSION
+VERSION=sdlttf_version
 
 AC_SUBST(MAJOR_VERSION)
 AC_SUBST(MINOR_VERSION)
@@ -25,6 +31,8 @@
 AC_SUBST(BINARY_AGE)
 AC_SUBST(VERSION)
 
+AC_CONFIG_MACRO_DIR([acinclude])
+
 # libtool versioning
 LT_INIT([win32-dll])
 
@@ -42,7 +50,7 @@
 AC_CANONICAL_HOST
 
 dnl Setup for automake
-AM_INIT_AUTOMAKE(SDL_ttf, $VERSION)
+AM_INIT_AUTOMAKE([foreign])
 
 dnl Check for tools
 AC_PROG_LIBTOOL
