--- /tmp/configure.ac	2008-06-22 14:14:59.000000000 +0200
+++ binutils-2.18.50.0.7/configure.ac	2008-06-22 14:15:30.000000000 +0200
@@ -561,7 +561,7 @@
     noconfigdirs="$noconfigdirs target-libffi target-qthreads"
     libgloss_dir=arm
     ;;
-  arm*-*-linux-gnueabi)
+  arm*-*-linux-gnueabi | arm*-*-linux-uclibceabi)
     noconfigdirs="$noconfigdirs target-qthreads"
     noconfigdirs="$noconfigdirs target-libobjc"
     case ${with_newlib} in
--- /tmp/configure	2008-06-22 14:17:11.000000000 +0200
+++ binutils-2.18.50.0.7/configure	2008-06-22 14:17:56.000000000 +0200
@@ -2307,7 +2307,7 @@
     noconfigdirs="$noconfigdirs target-libffi target-qthreads"
     libgloss_dir=arm
     ;;
-  arm*-*-linux-gnueabi)
+  arm*-*-linux-gnueabi | arm*-*-linux-uclibceabi)
     noconfigdirs="$noconfigdirs target-qthreads"
     noconfigdirs="$noconfigdirs target-libobjc"
     case ${with_newlib} in
