From 4af9cc9e93e5cb09583c0fedeccae3fa195dbe7b Mon Sep 17 00:00:00 2001
From: Etienne Cordonnier <ecordonnier@snap.com>
Date: Fri, 3 Mar 2023 13:26:08 +0100
Subject: [PATCH 08/15] libcutils: modifications to make it build in yocto
 environment
---
 debian/libcutils.mk | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/debian/libcutils.mk b/debian/libcutils.mk
index 7cd18d87..bb45fb57 100644
--- a/debian/libcutils.mk
+++ b/debian/libcutils.mk
@@ -1,3 +1,4 @@
+include ../../rules_yocto.mk
 NAME = libcutils
 
 # copied from libcutils/Android.bp
@@ -50,14 +51,15 @@ CPPFLAGS += \
             -Iinclude \
 
 LDFLAGS += -shared -Wl,-soname,$(NAME).so.0 \
-           -Wl,-rpath=/usr/lib/$(DEB_HOST_MULTIARCH)/android -lpthread -L. -llog -lbase
+           -Wl,-rpath=/usr/lib/$(DEB_HOST_MULTIARCH)/android -lpthread -L$(OUT_DIR)/usr/lib/$(DEB_HOST_MULTIARCH)/android/ -llog -lbase 
 
 build: $(COBJECTS) $(CXXOBJECTS)
-	$(CXX) $^ -o $(NAME).so.0 $(LDFLAGS)
-	ln -s $(NAME).so.0 $(NAME).so
+	$(CXX) $^ -o $(OUT_DIR)/usr/lib/$(DEB_HOST_MULTIARCH)/android/$(NAME).so.0 $(LDFLAGS)	
+	ln -sf $(NAME).so.0 $(OUT_DIR)/usr/lib/$(DEB_HOST_MULTIARCH)/android/$(NAME).so
+	cp -r libcutils/include/cutils $(OUT_DIR)/usr/include/	
 
 clean:
-	$(RM) $(CXXOBJECTS) $(COBJECTS) $(NAME).so*
+	$(RM) $(CXXOBJECTS) $(COBJECTS) $(OUT_DIR)/usr/lib/$(DEB_HOST_MULTIARCH)/android/$(NAME).so*
 
 $(COBJECTS): %.o: %.c
 	$(CC) $< -o $@ $(CFLAGS) $(CPPFLAGS)
-- 
2.36.1.vfs.0.0

