From 25bd3456dcd539b89648273152e405314287f4f8 Mon Sep 17 00:00:00 2001
From: Etienne Cordonnier <ecordonnier@snap.com>
Date: Mon, 13 Mar 2023 13:46:14 +0100
Subject: [PATCH 2/2] img2simg: Fix wrong rpath and re-add libbase

Signed-off-by: Etienne Cordonnier <ecordonnier@snap.com>
---
 debian/img2simg.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/debian/img2simg.mk b/debian/img2simg.mk
index d1b9b927..459af454 100644
--- a/debian/img2simg.mk
+++ b/debian/img2simg.mk
@@ -3,7 +3,7 @@ NAME = img2simg
 SOURCES = img2simg.cpp
 SOURCES := $(foreach source, $(SOURCES), libsparse/$(source))
 CPPFLAGS += -Ilibsparse/include -Iinclude -std=gnu++17
-LDFLAGS += -Wl,-rpath=$(OUT_DIR)/usr/lib/$(DEB_HOST_MULTIARCH)/android \
+LDFLAGS += -Wl,-rpath='$$ORIGIN/../lib/android' -lbase \
            -Wl,-rpath-link=$(OUT_DIR)/usr/lib/$(DEB_HOST_MULTIARCH)/android/ \
            -L$(OUT_DIR)/usr/lib/$(DEB_HOST_MULTIARCH)/android/ -lsparse
 
-- 
2.36.1.vfs.0.0

