From ca6400743a86fa334936d9984dcd40e90bb06e60 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 19 Apr 2025 00:09:42 -0700
Subject: [PATCH] cargo.toml: Update bitvec to use radium 1.x

Upstream-Status: Submitted [https://github.com/ferrilab/bitvec/pull/220]

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Signed-off-by: Leon Anavi <leon.anavi@konsulko.com>
---
 Cargo.lock | 10 +++++++---
 Cargo.toml |  2 ++
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index 6f26c8a..91ceade 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -55,7 +55,7 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "1bc2832c24239b0141d5674bb9174f9d68a8b5b3f2753311927c172ca46f7e9c"
 dependencies = [
  "funty",
- "radium",
+ "radium 1.1.0",
  "tap",
  "wyz",
 ]
@@ -469,6 +469,7 @@ dependencies = [
  "percent-encoding",
  "pyo3",
  "pyo3-build-config",
+ "radium 1.1.0",
  "regex",
  "serde",
  "serde_json",
@@ -569,9 +570,12 @@ checksum = "74765f6d916ee2faa39bc8e68e4f3ed8949b48cccdac59983d287a7cb71ce9c5"
 
 [[package]]
 name = "radium"
-version = "0.7.0"
+version = "1.1.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "dc33ff2d4973d518d823d61aa239014831e521c75da58e3df4840d3f47749d09"
+checksum = "db0b76288902db304c864a12046b73d2d895cc34a4bb8137baaeebe9978a072c"
+dependencies = [
+ "cfg-if",
+]
 
 [[package]]
 name = "regex"
diff --git a/Cargo.toml b/Cargo.toml
index 1fb5dcc..4949c8a 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -25,6 +25,8 @@ include = [
 rust-version = "1.88"
 
 [dependencies]
+#bitvec = { git = "https://github.com/alexanderkjall/bitvec", branch = "upgrade-radium-to-1" }
+radium = "1"
 # TODO it would be very nice to remove the "py-clone" feature as it can panic,
 # but needs a bit of work to make sure it's not used in the codebase
 pyo3 = { version = "0.28", features = ["generate-import-lib", "num-bigint", "py-clone", "smallvec"] }
-- 
2.47.3

