From dd1406e40700df60a055db24d7358480b1fbc5bf Mon Sep 17 00:00:00 2001
From: Mike Blumenkrantz <zmike@samsung.com>
Date: Mon, 3 Mar 2014 17:38:55 -0500
Subject: [PATCH] remove another elm 1.8 requirement

Upstream-Status: backport dd1406e40700df60a055db24d7358480b1fbc5bf
---
 configure.ac | 1 -
 1 file changed, 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 999df79..26f6775 100644
--- a/configure.ac
+++ b/configure.ac
@@ -579,7 +579,6 @@ eina >= ${efl_version} \
 eldbus >= ${efl_version} \
 eio >= ${efl_version} \
 elementary >= ${elm_version} \
-elementary < 1.8.99 \
 emotion >= ${efl_version} \
 $udisks_mount \
 $eeze_mount \
-- 
1.9.0

