Index: e/config/illume/e.src
===================================================================
--- e.orig/config/illume/e.src	2009-01-30 23:34:37.000000000 +0100
+++ e/config/illume/e.src	2009-01-30 23:34:54.000000000 +0100
@@ -665,7 +665,7 @@
   value "display_res_height" int: 1;
   value "display_res_hz" int: 0;
   value "display_res_rotation" int: 0;
-  value "screensaver_enable" int: 1;
+  value "screensaver_enable" int: 0;
   value "screensaver_timeout" int: 30;
   value "screensaver_interval" int: 5;
   value "screensaver_blanking" int: 2;
