--- /tmp/qlibraryinfo.cpp	2008-08-02 17:16:41.000000000 +0200
+++ qt-embedded-linux-opensource-src-4.4.1/src/corelib/global/qlibraryinfo.cpp	2008-08-02 17:17:42.000000000 +0200
@@ -47,6 +47,7 @@
 QT_BEGIN_NAMESPACE
 extern QString qmake_libraryInfoFile();
 QT_END_NAMESPACE
+#include <stdlib.h>
 #else
 # include "qcoreapplication.h"
 #endif
@@ -113,6 +114,10 @@
 #ifdef QT_BUILD_QMAKE
     if(!QFile::exists(qtconfig))
         qtconfig = qmake_libraryInfoFile();
+    if (!QFile::exists(qtconfig)) {
+        QByteArray config = getenv("QT_CONF_PATH");
+        qtconfig = QFile::decodeName(config);
+    }
 #else
     if (!QFile::exists(qtconfig) && QCoreApplication::instance()) {
 #ifdef Q_OS_MAC
