Index: qtopia-core-opensource-src-4.3.1/qmake/generators/unix/unixmake2.cpp
===================================================================
--- qtopia-core-opensource-src-4.3.1.orig/qmake/generators/unix/unixmake2.cpp	2007-10-07 13:45:26.000000000 +0200
+++ qtopia-core-opensource-src-4.3.1/qmake/generators/unix/unixmake2.cpp	2007-10-07 13:57:09.000000000 +0200
@@ -416,7 +416,7 @@
                 t << "\n\t" << mkdir_p_asstring(destdir) << "\n\t";
             if(!project->isEmpty("QMAKE_PRE_LINK"))
                 t << var("QMAKE_PRE_LINK") << "\n\t";
-            t << "$(LINK) $(LFLAGS) -o $(TARGET) " << incr_deps << " " << incr_objs << " $(OBJCOMP) $(LIBS)";
+            t << "$(LINK) -o $(TARGET) " << incr_deps << " " << incr_objs << " $(OBJCOMP) $(LIBS) $(LFLAGS)";
             if(!project->isEmpty("QMAKE_POST_LINK"))
                 t << "\n\t" << var("QMAKE_POST_LINK");
             t << endl << endl;
@@ -430,7 +430,7 @@
                 t << mkdir_p_asstring(destdir) << "\n\t";
             if(!project->isEmpty("QMAKE_PRE_LINK"))
                 t << var("QMAKE_PRE_LINK") << "\n\t";
-            t << "$(LINK) $(LFLAGS) -o $(TARGET) $(OBJECTS) $(OBJCOMP) $(LIBS)";
+            t << "$(LINK) -o $(TARGET) $(OBJECTS) $(OBJCOMP) $(LIBS) $(LFLAGS)";
             if(!project->isEmpty("QMAKE_POST_LINK"))
                 t << "\n\t" << var("QMAKE_POST_LINK");
             t << endl << endl;
