Index: fuse-2.8.6/util/fusermount.c
===================================================================
--- fuse-2.8.6.orig/util/fusermount.c	2011-09-13 00:23:14.000000000 -0700
+++ fuse-2.8.6/util/fusermount.c	2011-09-25 00:23:53.051019038 -0700
@@ -7,6 +7,7 @@
 */
 /* This program does the mounting and unmounting of FUSE filesystems */
 
+#define _GNU_SOURCE
 #include <config.h>
 
 #include "mount_util.h"
