Index: evince-2.30.0/gnome-doc-utils.make
===================================================================
--- evince-2.30.0.orig/gnome-doc-utils.make	2010-03-29 11:52:47.000000000 -0700
+++ evince-2.30.0/gnome-doc-utils.make	2010-08-17 15:20:22.597646001 -0700
@@ -132,13 +132,12 @@ _DOC_ABS_SRCDIR = @abs_srcdir@
 
 _xml2po ?= `which xml2po`
 _xml2po_mode = $(if $(DOC_ID),mallard,docbook)
-
-_db2html ?= `$(PKG_CONFIG) --variable db2html gnome-doc-utils`
-_db2omf  ?= `$(PKG_CONFIG) --variable db2omf gnome-doc-utils`
-_malrng  ?= `$(PKG_CONFIG) --variable malrng gnome-doc-utils`
-_chunks  ?= `$(PKG_CONFIG) --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/chunks.xsl
-_credits ?= `$(PKG_CONFIG) --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/credits.xsl
-_ids ?= `$(PKG_CONFIG) --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/ids.xsl
+_db2html ?= ${PKG_CONFIG_SYSROOT_DIR}`$(PKG_CONFIG) --variable db2html gnome-doc-utils`
+_db2omf  ?= ${PKG_CONFIG_SYSROOT_DIR}`$(PKG_CONFIG) --variable db2omf gnome-doc-utils`
+_malrng  ?= ${PKG_CONFIG_SYSROOT_DIR}`$(PKG_CONFIG) --variable malrng gnome-doc-utils`
+_chunks  ?= ${PKG_CONFIG_SYSROOT_DIR}`$(PKG_CONFIG) --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/chunks.xsl
+_credits ?= ${PKG_CONFIG_SYSROOT_DIR}`$(PKG_CONFIG) --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/credits.xsl
+_ids ?= ${PKG_CONFIG_SYSROOT_DIR}`$(PKG_CONFIG) --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/ids.xsl
 
 if ENABLE_SK
 _ENABLE_SK = true

