From a574c6ffa818e91772a3bd37649f552b62cd416e Mon Sep 17 00:00:00 2001
From: Dirk Lemstra <dirk@lemstra.org>
Date: Mon, 12 May 2025 22:23:48 +0200
Subject: [PATCH] Added missing return.

CVE: CVE-2025-53015
Upstream-Status: Backport [https://github.com/ImageMagick/ImageMagick/commit/38631605e6ab744548a561797472cf8648bcfe26]
Signed-off-by: Gyorgy Sarvari <skandigraun@gmail.com>
---
 MagickCore/profile.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/MagickCore/profile.c b/MagickCore/profile.c
index 85c180118..a68e54f14 100644
--- a/MagickCore/profile.c
+++ b/MagickCore/profile.c
@@ -2581,6 +2581,7 @@ static void GetXmpNumeratorAndDenominator(double value,
     {
       *numerator = (unsigned long) value;
       *denominator = 1;
+      return;
     }
   *numerator=1;
   df=1.0;
