diff -ruN libnl-1.1/lib/route/link/vlan.c libnl-1.1-new/lib/route/link/vlan.c
--- libnl-1.1/lib/route/link/vlan.c	2008-01-14 18:48:45.000000000 +0300
+++ libnl-1.1-new/lib/route/link/vlan.c	2009-01-30 10:55:09.000000000 +0300
@@ -26,7 +26,9 @@
 #include <netlink/route/link/info-api.h>
 #include <netlink/route/link/vlan.h>
 
+#ifndef VLAN_FLAG_REORDER_HDR
 #include <linux/if_vlan.h>
+#endif
 
 /** @cond SKIP */
 #define VLAN_HAS_ID		(1<<0)
