diff -pruN libnl-1.1.orig/include/netlink/handlers.h libnl-1.1/include/netlink/handlers.h
--- libnl-1.1.orig/include/netlink/handlers.h	2008-01-14 18:48:45.000000000 +0300
+++ libnl-1.1/include/netlink/handlers.h	2011-04-12 19:23:16.603292527 +0400
@@ -120,6 +120,7 @@ extern struct nl_cb *	nl_cb_alloc(enum n
 extern struct nl_cb *	nl_cb_clone(struct nl_cb *);
 extern struct nl_cb *	nl_cb_get(struct nl_cb *);
 extern void		nl_cb_put(struct nl_cb *);
+struct ucred;
 
 extern int  nl_cb_set(struct nl_cb *, enum nl_cb_type, enum nl_cb_kind,
 		      nl_recvmsg_msg_cb_t, void *);
diff -pruN libnl-1.1.orig/include/netlink/netlink.h libnl-1.1/include/netlink/netlink.h
--- libnl-1.1.orig/include/netlink/netlink.h	2008-01-14 18:48:45.000000000 +0300
+++ libnl-1.1/include/netlink/netlink.h	2011-04-12 19:23:51.370292658 +0400
@@ -50,6 +50,8 @@ extern int			nl_send_auto_complete(struc
 extern int			nl_send_simple(struct nl_handle *, int, int,
 					       void *, size_t);
 
+struct ucred;
+
 /* Receive */
 extern int			nl_recv(struct nl_handle *,
 					struct sockaddr_nl *, unsigned char **,
