Index: libnl-1.1/include/netlink-local.h
===================================================================
--- libnl-1.1.orig/include/netlink-local.h	2009-06-18 15:28:32.614209645 +0400
+++ libnl-1.1/include/netlink-local.h	2009-06-18 15:28:44.094190518 +0400
@@ -26,6 +26,7 @@
 #include <sys/socket.h>
 #include <inttypes.h>
 #include <assert.h>
+#include <limits.h>
 
 #include <arpa/inet.h>
 #include <netdb.h>
