diff -urN libnl-1.1.orig/lib/Makefile libnl-1.1/lib/Makefile
--- libnl-1.1.orig/lib/Makefile	2008-01-14 15:48:45.000000000 +0000
+++ libnl-1.1/lib/Makefile	2009-04-06 10:16:21.000000000 +0000
@@ -48,7 +48,7 @@
 
 $(OUT_SLIB): ../Makefile.opts $(OBJ)
 	@echo "  LD $(OUT_SLIB)"; \
-	$(CC) -shared -Wl,-soname,libnl.so.1 -o $(OUT_SLIB) $(OBJ) $(LIBNL_LIB) -lc
+	$(CC) $(LDFLAGS) -shared -Wl,-soname,libnl.so.1 -o $(OUT_SLIB) $(OBJ) $(LIBNL_LIB) -lc
 	@echo "  LN $(OUT_SLIB) $(LN1_SLIB)"; \
 	rm -f $(LN1_SLIB) ; $(LN) -s $(OUT_SLIB) $(LN1_SLIB)
 	@echo "  LN $(LN1_SLIB) $(LN_SLIB)"; \
