--- links-2.1pre17/cookies.c.pix	2005-05-15 23:05:10.000000000 +0800
+++ links-2.1pre17/cookies.c	2005-05-15 23:17:21.000000000 +0800
@@ -41,7 +41,7 @@
 
 void free_cookie(struct cookie *c)
 {
-	mem_free(c->name);
+	if (c->value) mem_free(c->name);
 	if (c->value) mem_free(c->value);
 	if (c->server) mem_free(c->server);
 	if (c->path) mem_free(c->path);
@@ -355,7 +355,7 @@
 
 inv:
 		free_cookie(cookie);
-		free(cookie);
+		mem_free(cookie);
 	}
 	fclose(fp);
 }
