From aaa6a81cfb20f9b3c25aa22e5a472c0f2867f72c Mon Sep 17 00:00:00 2001
From: Claude Bing <cbing@cybernetics.com>
Date: Tue, 9 Nov 2021 13:11:42 -0500
Subject: [PATCH 11/11] opcache/jit: use minilua in sysroot

Signed-off-by: Claude Bing <cbing@cybernetics.com>
---
 ext/opcache/jit/Makefile.frag | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/ext/opcache/jit/Makefile.frag b/ext/opcache/jit/Makefile.frag
index b3af5b290a..f2e737e463 100644
--- a/ext/opcache/jit/Makefile.frag
+++ b/ext/opcache/jit/Makefile.frag
@@ -1,9 +1,6 @@
 
-$(builddir)/minilua: $(srcdir)/jit/dynasm/minilua.c
-	$(CC) $(srcdir)/jit/dynasm/minilua.c -lm -o $@
-
-$(builddir)/jit/zend_jit_x86.c: $(srcdir)/jit/zend_jit_x86.dasc $(srcdir)/jit/dynasm/*.lua $(builddir)/minilua
-	$(builddir)/minilua $(srcdir)/jit/dynasm/dynasm.lua  $(DASM_FLAGS) -o $@ $(srcdir)/jit/zend_jit_x86.dasc
+$(builddir)/jit/zend_jit_x86.c: $(srcdir)/jit/zend_jit_x86.dasc $(srcdir)/jit/dynasm/*.lua
+	@minilua $(srcdir)/jit/dynasm/dynasm.lua  $(DASM_FLAGS) -o $@ $(srcdir)/jit/zend_jit_x86.dasc
 
 $(builddir)/jit/zend_jit.lo: \
 	$(builddir)/jit/zend_jit_x86.c \
-- 
2.25.1

