From 617d01ef2ac2f436cd8e06555b608db56079b851 Mon Sep 17 00:00:00 2001
From: Paul Eggleton <paul.eggleton@linux.intel.com>
Date: Tue, 17 Jul 2012 11:31:54 +0100

---
 configure | 2 --
 1 file changed, 2 deletions(-)

diff --git a/configure b/configure
index 1f1ef32..392fb76 100755
--- a/configure
+++ b/configure
@@ -11642,8 +11642,6 @@ $as_echo "#define HAVE_LIBNSL 1" >>confdefs.h
 
 
 
-  unset ac_cv_func_dlopen
-  unset ac_cv_func___dlopen
   unset found
 
   ac_fn_c_check_func "$LINENO" "dlopen" "ac_cv_func_dlopen"
