---
 sapi/cli/config.m4 |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- php-5.4.14.orig/sapi/cli/config.m4
+++ php-5.4.14/sapi/cli/config.m4
@@ -34,11 +34,11 @@ if test "$PHP_CLI" != "no"; then
     BUILD_CLI="\$(LIBTOOL) --mode=link \$(CC) -export-dynamic \$(CFLAGS_CLEAN) \$(EXTRA_CFLAGS) \$(EXTRA_LDFLAGS_PROGRAM) \$(LDFLAGS) \$(PHP_RPATHS) \$(PHP_GLOBAL_OBJS) \$(PHP_BINARY_OBJS) \$(PHP_CLI_OBJS) \$(EXTRA_LIBS) \$(ZEND_EXTRA_LIBS) -o \$(SAPI_CLI_PATH)"
     ;;
   esac
 
   dnl Set executable for tests
-  PHP_EXECUTABLE="\$(top_builddir)/\$(SAPI_CLI_PATH)"
+  PHP_EXECUTABLE="${PHP_NATIVE_DIR}/php"
   PHP_SUBST(PHP_EXECUTABLE)
 
   dnl Expose to Makefile
   PHP_SUBST(SAPI_CLI_PATH)
   PHP_SUBST(BUILD_CLI)
