---
 configure.in |   12 ++++++++----
 1 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/configure.in b/configure.in
index eb3e5e8..11257c9 100644
--- a/configure.in
+++ b/configure.in
@@ -181,8 +181,9 @@ yes)   AC_MSG_RESULT(yes)
        ipv6=no
        ;;
   esac ],
-
-  AC_TRY_RUN([ /* AF_INET6 available check */
+[
+  if test x"$cross_compiling" != "xyes"; then
+    AC_TRY_RUN([ /* AF_INET6 avalable check */
 #include <sys/types.h>
 #include <sys/socket.h>
 main()
@@ -201,7 +202,10 @@ main()
   ipv6=no],
 [ AC_MSG_RESULT(no)
   ipv6=no]
-))
+  )
+else
+  AC_MSG_FAILURE([Unable to check for ipv6 when crosscompiling, please specify.])
+fi])
 
 ipv6type=unknown
 ipv6lib=none
@@ -316,7 +320,7 @@ if test "$ipv6" = "yes" -a "$ipv6lib" != "none"; then
 fi
 
 
-if test "$ipv6" = "yes"; then
+if test x"$cross_compiling" != "xyes" -a "$ipv6" = "yes"; then
 	#
 	# XXX - on Tru64 UNIX 5.1, there is no "getaddrinfo()"
 	# function in libc; there are "ngetaddrinfo()" and
