---
 configure |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/configure b/configure
index cb51d19..73d51af 100755
--- a/configure
+++ b/configure
@@ -2691,13 +2691,13 @@ if test "$ac_test_CFLAGS" = set; then
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g"
   else
     CFLAGS="-g"
   fi
 else
   if test "$GCC" = yes; then
-    CFLAGS="-O2"
+    CFLAGS=""
   else
     CFLAGS=
   fi
@@ -2830,7 +2830,7 @@ ac_compiler_gnu=$ac_cv_c_compiler_gnu
     if test "$GCC" = yes ; then
 	    if test "$SHLICC2" = yes ; then
 		    ac_cv_lbl_gcc_vers=2
-		    V_CCOPT="-O2"
+		    V_CCOPT=""
 	    else
 		    { echo "$as_me:$LINENO: checking gcc version" >&5
 echo $ECHO_N "checking gcc version... $ECHO_C" >&6; }
@@ -2847,7 +2847,7 @@ fi
 		    { echo "$as_me:$LINENO: result: $ac_cv_lbl_gcc_vers" >&5
 echo "${ECHO_T}$ac_cv_lbl_gcc_vers" >&6; }
 		    if test $ac_cv_lbl_gcc_vers -gt 1 ; then
-			    V_CCOPT="-O2"
+			    V_CCOPT=""
 		    fi
 	    fi
     else
