--- ZeroTierOne-1.2.10/make-linux.mk	2018-05-11 22:40:22.000000000 +0200
+++ ZeroTierOne-1.2.10-oe/make-linux.mk	2018-05-26 20:54:41.091647361 +0200
@@ -63,13 +63,13 @@
 	# C25519 in particular is almost UNUSABLE in -O0 even on a 3ghz box!
 node/Salsa20.o node/SHA512.o node/C25519.o node/Poly1305.o: CXXFLAGS=-Wall -O2 -g -pthread $(INCLUDES) $(DEFS)
 else
-	CFLAGS?=-O3 -fstack-protector -fPIE
+	CFLAGS?=-O3 -fstack-protector
 	override CFLAGS+=-Wall -Wno-deprecated -pthread $(INCLUDES) -DNDEBUG $(DEFS)
-	CXXFLAGS?=-O3 -fstack-protector -fPIE
+	CXXFLAGS?=-O3 -fstack-protector
 	override CXXFLAGS+=-Wall -Wno-deprecated -std=c++11 -pthread $(INCLUDES) -DNDEBUG $(DEFS)
-	LDFLAGS=-pie -Wl,-z,relro,-z,now
-	STRIP?=strip
-	STRIP+=--strip-all
+	LDFLAGS+= -Wl,-z,relro,-z,now
+	STRIP?=echo
+#	STRIP+=--strip-all
 endif
 
 ifeq ($(ZT_QNAP), 1)
