Index: gtk+-2.6.10/modules/engines/pixbuf/pixbuf-render.c
===================================================================
--- gtk+-2.6.10.orig/modules/engines/pixbuf/pixbuf-render.c	2005-08-18 14:11:00.000000000 +0000
+++ gtk+-2.6.10/modules/engines/pixbuf/pixbuf-render.c	2007-05-24 11:16:50.000000000 +0000
@@ -407,7 +407,7 @@
 			0, 0, rect.width, rect.height,
 			dest_x - rect.x, dest_y - rect.y, 
 			x_scale, y_scale,
-			GDK_INTERP_BILINEAR);
+			GDK_INTERP_NEAREST);
 
       g_object_unref (partial_src);
 
