From 738d48e46cbebfa22cbfc400ce2a7ab320f07359 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Sat, 24 Jun 2023 15:26:27 +0800
Subject: [PATCH] Makefile: remove help2man dependency

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 Makefile | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 0343224..f493778 100644
--- a/Makefile
+++ b/Makefile
@@ -27,13 +27,11 @@ include Make.rules
 
 EFISIGNED = $(patsubst %.efi,%-signed.efi,$(EFIFILES))
 
-all: $(EFISIGNED) $(BINARIES) $(MANPAGES) noPK.auth $(KEYAUTH) \
+all: $(EFISIGNED) $(BINARIES) noPK.auth $(KEYAUTH) \
 	$(KEYUPDATEAUTH) $(KEYBLACKLISTAUTH) $(KEYHASHBLACKLISTAUTH)
 
 
 install: all
-	$(INSTALL) -m 755 -d $(MANDIR)
-	$(INSTALL) -m 644 $(MANPAGES) $(MANDIR)
 	$(INSTALL) -m 755 -d $(EFIDIR)
 	$(INSTALL) -m 755 $(EFIFILES) $(EFIDIR)
 	$(INSTALL) -m 755 -d $(BINDIR)
-- 
2.25.1

