--- cups-1.4.3/ppdc/Makefile.orig	2010-04-07 12:38:56.650327699 +0200
+++ cups-1.4.3/ppdc/Makefile	2010-04-07 12:40:05.842418876 +0200
@@ -243,8 +243,8 @@
 	$(CXX) $(ARCHFLAGS) $(LDFLAGS) -o genstrings genstrings.o \
 		libcupsppdc.a ../cups/libcups.a $(LIBGSSAPI) $(SSLLIBS) \
 		$(DNSSDLIBS) $(COMMONLIBS) $(LIBZ)
-	echo Generating localization strings...
-	./genstrings >sample.c
+#	echo Generating localization strings...
+#	./genstrings >sample.c
 
 
 #
@@ -261,9 +261,9 @@
 	$(CXX) $(ARCHFLAGS) $(LDFLAGS) -o ppdc-static ppdc.o libcupsppdc.a \
 		../cups/libcups.a $(LIBGSSAPI) $(SSLLIBS) $(DNSSDLIBS) \
 		$(COMMONLIBS) $(LIBZ)
-	echo Testing PPD compiler...
-	./ppdc-static -l en,fr -I ../data foo.drv
-	./ppdc-static -l en,fr -z -I ../data foo.drv
+#	echo Testing PPD compiler...
+#	./ppdc-static -l en,fr -I ../data foo.drv
+#	./ppdc-static -l en,fr -z -I ../data foo.drv
 
 
 #
@@ -291,15 +291,15 @@
 		$(COMMONLIBS) $(LIBZ)
 	echo Testing PPD importer...
 	$(RM) -r ppd ppd2 sample-import.drv
-	./ppdc-static -I ../data sample.drv
-	./ppdi-static -I ../data -o sample-import.drv ppd/*
-	./ppdc-static -I ../data -d ppd2 sample-import.drv
-	if diff -qr ppd ppd2; then \
-		echo PPD import OK; \
-	else \
-		echo PPD import FAILED; \
-		exit 1; \
-	fi
+#	./ppdc-static -I ../data sample.drv
+#	./ppdi-static -I ../data -o sample-import.drv ppd/*
+#	./ppdc-static -I ../data -d ppd2 sample-import.drv
+#	if diff -qr ppd ppd2; then \
+#		echo PPD import OK; \
+#	else \
+#		echo PPD import FAILED; \
+#		exit 1; \
+#	fi
 
 
 #
