Upstream-Status: Backport [a346c582cc0f34ac6c7c11251919c84553424efb]
Signed-off-by: Ross Burton <ross.burton@arm.com>

diff --git a/data/installed-tests/fwupd.sh b/data/installed-tests/fwupd.sh
index 5b1f5d22..eaa5ec7f 100755
--- a/data/installed-tests/fwupd.sh
+++ b/data/installed-tests/fwupd.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 exec 2>&1
 dirname=`dirname $0`
@@ -7,7 +7,7 @@ run_test()
 {
         if [ -f $dirname/$1 ]; then
                 $dirname/$1
-                rc=$?; if [[ $rc != 0 ]]; then exit $rc; fi
+                rc=$?; if [ $rc != 0 ]; then exit $rc; fi
         fi
 }
 
diff --git a/data/installed-tests/fwupdmgr.sh b/data/installed-tests/fwupdmgr.sh
index e3f7ee31..3d65395e 100755
--- a/data/installed-tests/fwupdmgr.sh
+++ b/data/installed-tests/fwupdmgr.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 exec 2>&1
 device=08d460be0f1f9f128413f816022a6439e0078018
@@ -13,47 +13,47 @@ error()
 # ---
 echo "Getting the list of remotes..."
 fwupdmgr get-remotes
-rc=$?; if [[ $rc != 0 ]]; then error $rc; fi
+rc=$?; if [ $rc != 0 ]; then error $rc; fi
 
 # ---
 echo "Enabling fwupd-tests remote..."
 fwupdmgr enable-remote fwupd-tests
-rc=$?; if [[ $rc != 0 ]]; then error $rc; fi
+rc=$?; if [ $rc != 0 ]; then error $rc; fi
 
 # ---
 echo "Update the device hash database..."
 fwupdmgr verify-update $device
-rc=$?; if [[ $rc != 0 ]]; then error $rc; fi
+rc=$?; if [ $rc != 0 ]; then error $rc; fi
 
 # ---
 echo "Getting devices (should be one)..."
 fwupdmgr get-devices --no-unreported-check
-rc=$?; if [[ $rc != 0 ]]; then error $rc; fi
+rc=$?; if [ $rc != 0 ]; then error $rc; fi
 
 # ---
 echo "Testing the verification of firmware..."
 fwupdmgr verify $device
-rc=$?; if [[ $rc != 0 ]]; then error $rc; fi
+rc=$?; if [ $rc != 0 ]; then error $rc; fi
 
 # ---
 echo "Getting updates (should be one)..."
 fwupdmgr --no-unreported-check --no-metadata-check get-updates
-rc=$?; if [[ $rc != 0 ]]; then error $rc; fi
+rc=$?; if [ $rc != 0 ]; then error $rc; fi
 
 # ---
 echo "Installing test firmware..."
 fwupdmgr update $device -y
-rc=$?; if [[ $rc != 0 ]]; then error $rc; fi
+rc=$?; if [ $rc != 0 ]; then error $rc; fi
 
 # ---
 echo "Getting updates (should be none)..."
 fwupdmgr --no-unreported-check --no-metadata-check get-updates
-rc=$?; if [[ $rc != 2 ]]; then error $rc; fi
+rc=$?; if [ $rc != 2 ]; then error $rc; fi
 
 # ---
 echo "Testing the verification of firmware (again)..."
 fwupdmgr verify $device
-rc=$?; if [[ $rc != 0 ]]; then error $rc; fi
+rc=$?; if [ $rc != 0 ]; then error $rc; fi
 
 if [ -z "$CI_NETWORK" ]; then
         echo "Skipping remaining tests due to CI_NETWORK not being set"
@@ -63,27 +63,27 @@ fi
 # ---
 echo "Downgrading to older release (requires network access)"
 fwupdmgr downgrade $device -y
-rc=$?; if [[ $rc != 0 ]]; then error $rc; fi
+rc=$?; if [ $rc != 0 ]; then error $rc; fi
 
 # ---
 echo "Downgrading to older release (should be none)"
 fwupdmgr downgrade $device
-rc=$?; if [[ $rc != 2 ]]; then error $rc; fi
+rc=$?; if [ $rc != 2 ]; then error $rc; fi
 
 # ---
 echo "Updating all devices to latest release (requires network access)"
 fwupdmgr --no-unreported-check --no-metadata-check --no-reboot-check update -y
-rc=$?; if [[ $rc != 0 ]]; then error $rc; fi
+rc=$?; if [ $rc != 0 ]; then error $rc; fi
 
 # ---
 echo "Getting updates (should be none)..."
 fwupdmgr --no-unreported-check --no-metadata-check get-updates
-rc=$?; if [[ $rc != 2 ]]; then error $rc; fi
+rc=$?; if [ $rc != 2 ]; then error $rc; fi
 
 # ---
 echo "Refreshing from the LVFS (requires network access)..."
 fwupdmgr refresh
-rc=$?; if [[ $rc != 0 ]]; then error $rc; fi
+rc=$?; if [ $rc != 0 ]; then error $rc; fi
 
 # success!
 exit 0
diff --git a/plugins/uefi-capsule/fwupd.grub.conf.in b/plugins/uefi-capsule/fwupd.grub.conf.in
index 92bd49a1..9c3a22f2 100755
--- a/plugins/uefi-capsule/fwupd.grub.conf.in
+++ b/plugins/uefi-capsule/fwupd.grub.conf.in
@@ -1,4 +1,4 @@
-#! /bin/bash
+#! /bin/sh
 # SPDX-License-Identifier: LGPL-2.1+
 set -e
 
@@ -8,7 +8,7 @@ set -e
 
 if [ -f @localstatedir@/lib/fwupd/uefi_capsule.conf ] &&
    ls /sys/firmware/efi/efivars/fwupd-*-0abba7dc-e516-4167-bbf5-4d9d1c739416 1>/dev/null 2>&1; then
-      source @localstatedir@/lib/fwupd/uefi_capsule.conf
+      . @localstatedir@/lib/fwupd/uefi_capsule.conf
       if [ "${EFI_PATH}" != "" ] && [ "${ESP}" != "" ]; then
       echo "Adding Linux Firmware Updater entry" >&2
 cat << EOF
