diff -Naur dbus-1.2.1.orig/dbus/dbus-transport.c dbus-1.2.1/dbus/dbus-transport.c
--- dbus-1.2.1.orig/dbus/dbus-transport.c	2008-09-16 15:10:02.000000000 +0000
+++ dbus-1.2.1/dbus/dbus-transport.c	2008-09-16 15:10:10.000000000 +0000
@@ -747,6 +747,9 @@
           dbus_bool_t allow;
           DBusCredentials *auth_identity;
           
+          // allow anonymous connections
+          _dbus_transport_set_allow_anonymous(transport, TRUE);
+
           auth_identity = _dbus_auth_get_identity (transport->auth);
           _dbus_assert (auth_identity != NULL);
           
