From aabfb6f78af8e337d3239142117ba303fce55e7e Mon Sep 17 00:00:00 2001
From: Dmitry Eremin-Solenikov <dbaryshkov@gmail.com>
Date: Thu, 22 Sep 2011 08:55:26 +0200
Subject: [PATCH] fix the parallel build regarding shared libraries.

Upstream-Status: Pending
---
 .../openssl-1.0.0e/Makefile.org                    |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile.org
index 3c7aea1..6326cd6 100644
--- a/Makefile.org
+++ b/Makefile.org
@@ -243,13 +243,13 @@ build_libs: build_crypto build_ssl build_engines
 
 build_crypto:
 	@dir=crypto; target=all; $(BUILD_ONE_CMD)
-build_ssl:
+build_ssl: build_crypto
 	@dir=ssl; target=all; $(BUILD_ONE_CMD)
-build_engines:
+build_engines: build_crypto
 	@dir=engines; target=all; $(BUILD_ONE_CMD)
-build_apps:
+build_apps: build_crypto build_ssl
 	@dir=apps; target=all; $(BUILD_ONE_CMD)
-build_tests:
+build_tests: build_crypto build_ssl
 	@dir=test; target=all; $(BUILD_ONE_CMD)
 build_tools:
 	@dir=tools; target=all; $(BUILD_ONE_CMD)
-- 
1.6.6.1

