Index: uClibc/ldso/ldso/arm/resolve.S
===================================================================
--- uClibc.orig/ldso/ldso/arm/resolve.S	2009-03-20 12:03:32.000000000 -0700
+++ uClibc/ldso/ldso/arm/resolve.S	2009-03-20 12:04:23.000000000 -0700
@@ -97,7 +97,6 @@
 
  .text
  .align 4      @ 16 byte boundary and there are 32 bytes below (arm case)
- #if !defined(__thumb__) || defined(__thumb2__)
  .arm
  .globl _dl_linux_resolve
  .type _dl_linux_resolve,%function
@@ -129,7 +128,7 @@
 #else
 	mov pc,ip
 #endif
-#else
+#if 0
        @ In the thumb case _dl_linux_resolver is thumb.  If a bl is used
        @ from arm code the linker will insert a stub call which, with
        @ binutils 2.16, is not PIC.  Since this code is accessed by an
