Index: uClibc/test/math/libm-test.inc
===================================================================
--- uClibc/test/math/libm-test.inc	(revision 23784)
+++ uClibc/test/math/libm-test.inc	(working copy)
@@ -3414,7 +3414,6 @@
 }
 
 
-#if 0
 static void
 log2_test (void)
 {
@@ -3444,7 +3443,6 @@
 
   END (log2);
 }
-#endif
 
 
 static void
@@ -4967,9 +4965,7 @@
   log_test ();
   log10_test ();
   log1p_test ();
-#if 0
   log2_test ();
-#endif
   logb_test ();
   modf_test ();
   ilogb_test ();
