From e8ecb307b6402b39d033e20f4df4f3374e7c0685 Mon Sep 17 00:00:00 2001
From: Li xin <lixin.fnst@cn.fujitsu.com>
Date: Fri, 21 Nov 2014 12:02:51 +0900
Subject: [PATCH] usr/iscsi_sysfs.c : add include files

Signed-off-by: Li Xin <lixin.fnst@cn.fujitsu.com>
---
 usr/iscsi_sysfs.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/usr/iscsi_sysfs.c b/usr/iscsi_sysfs.c
index cb9a5a1..f6bd78a 100644
--- a/usr/iscsi_sysfs.c
+++ b/usr/iscsi_sysfs.c
@@ -14,6 +14,8 @@
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
  * General Public License for more details.
  */
+#include <sys/types.h>
+#include <sys/stat.h>
 #include <fcntl.h>
 #include <unistd.h>
 #include <stdlib.h>
-- 
1.8.4.2

