From 857c38eb8201e2a34b524fcabea21b1accbdb288 Mon Sep 17 00:00:00 2001
From: Philip Balister <philip@opensdr.com>
Date: Fri, 13 May 2011 12:26:14 -0400
Subject: [PATCH] volk : Remove all traces of volk from configure for OE builds.

Signed-off-by: Philip Balister <philip@opensdr.com>
---
 Makefile.common |    4 ++--
 configure.ac    |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile.common b/Makefile.common
index 20b8a78..aeda1dd 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -92,8 +92,8 @@ GRUEL_INCLUDES = @gruel_INCLUDES@
 GRUEL_LA = @gruel_LA@
 
 # How to link in the VOLK library from inside the tree
-VOLK_INCLUDES = @volk_INCLUDES@
-VOLK_LA = @volk_LA@
+#VOLK_INCLUDES = @volk_INCLUDES@
+#VOLK_LA = @volk_LA@
 
 # How to link in the USRP library from inside the tree
 USRP_INCLUDES = @usrp_INCLUDES@
diff --git a/configure.ac b/configure.ac
index 73ebbd6..e5cf7a2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -349,7 +349,7 @@ AC_ARG_ENABLE(
 
 build_dirs="config"
 GRC_GRUEL                       dnl must come first
-GRC_VOLK
+dnl GRC_VOLK
 GRC_GCELL
 GRC_GNURADIO_CORE
 GRC_USRP
-- 
1.7.3.4

