diff --git a/src/interfaces/ecpg/Makefile b/src/interfaces/ecpg/Makefile
index dcd578f..1428423 100644
--- a/src/interfaces/ecpg/Makefile
+++ b/src/interfaces/ecpg/Makefile
@@ -8,7 +8,6 @@ all install installdirs uninstall dep depend distprep:
 	$(MAKE) -C ecpglib $@
 	$(MAKE) -C compatlib $@
 	$(MAKE) -C preproc $@
-	$(MAKE) -C test $@
 
 clean distclean maintainer-clean:
 	-$(MAKE) -C include $@
