--- gcc-3.4.6/gcc/collect2.c	2008-10-04 18:17:17.796750393 +0400
+++ gcc-3.4.6/gcc/collect2.new	2008-10-04 18:24:10.120748711 +0400
@@ -1534,7 +1534,7 @@ collect_execute (const char *prog, char 
   if (redir)
     {
       /* Open response file.  */
-      redir_handle = open (redir, O_WRONLY | O_TRUNC | O_CREAT);
+      redir_handle = open (redir, O_WRONLY | O_TRUNC | O_CREAT, S_IWUSR);
 
       /* Duplicate the stdout and stderr file handles
 	 so they can be restored later.  */
