From 58db99e98f615d79ea90cac8f4bcf11c94e3e7c7 Mon Sep 17 00:00:00 2001
From: Marcin Juszkiewicz <marcin.juszkiewicz@linaro.org>
Date: Thu, 10 Jan 2013 12:42:19 +0100
Subject: [PATCH] configure: enable pic for AArch64

Signed-off-by: Marcin Juszkiewicz <marcin.juszkiewicz@linaro.org>

Upstream-Status: Backport
---
 configure |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- git.orig/configure
+++ git/configure
@@ -2390,11 +2390,11 @@ EOF
 
 check_host_cflags -std=c99
 check_host_cflags -Wall
 
 case "$arch" in
-    alpha|ia64|mips|parisc|sparc)
+    alpha|ia64|mips|parisc|sparc|aarch64)
         spic=$shared
     ;;
     x86)
         subarch="x86_32"
         check_cc <<EOF && subarch="x86_64"
