From 7643b6615aab4de290ead6c1a6b56f10a82bbe6f Mon Sep 17 00:00:00 2001
From: Roy Marples <roy@marples.name>
Date: Sat, 4 Jul 2020 13:21:09 +0100
Subject: [PATCH] Fix compile without IPv6 on systems that do not define ALIGN

Upstream-Status: backport

---
 src/dhcpcd.h | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/src/dhcpcd.h b/src/dhcpcd.h
index 61a6a081..d9719c0a 100644
--- a/src/dhcpcd.h
+++ b/src/dhcpcd.h
@@ -96,7 +96,6 @@ TAILQ_HEAD(if_head, interface);
 
 #include "privsep.h"
 
-#ifdef INET6
 /* dhcpcd requires CMSG_SPACE to evaluate to a compile time constant. */
 #if defined(__QNX) || \
 	(defined(__NetBSD_Version__) && __NetBSD_Version__ < 600000000)
@@ -113,10 +112,6 @@ TAILQ_HEAD(if_head, interface);
 #define	CMSG_SPACE(len)	(ALIGN(sizeof(struct cmsghdr)) + ALIGN(len))
 #endif
 
-#define IP6BUFLEN	(CMSG_SPACE(sizeof(struct in6_pktinfo)) + \
-			CMSG_SPACE(sizeof(int)))
-#endif
-
 struct passwd;
 
 struct dhcpcd_ctx {
