diff -uNr gtk+-3.2.3.orig/gtk/Makefile.am gtk+-3.2.3/gtk/Makefile.am
--- gtk+-3.2.3.orig/gtk/Makefile.am	2011-12-19 14:58:28.000000000 +0100
+++ gtk+-3.2.3/gtk/Makefile.am	2012-07-18 02:01:59.042628402 +0200
@@ -942,7 +942,7 @@
 
 if USE_QUARTZ
 install-mac-key-theme:
-	$(mkdir_p) $(DESTDIR)$(datadir)/themes/Mac/gtk-3.0
+	$(MKDIR_P) $(DESTDIR)$(datadir)/themes/Mac/gtk-3.0
 	$(INSTALL_DATA) $(srcdir)/gtk-keys.css.mac $(DESTDIR)$(datadir)/themes/Mac/gtk-3.0/gtk-keys.css
 uninstall-mac-key-theme:
 	rm -f $(DESTDIR)$(datadir)/themes/Mac/gtk-3.0/gtk-keys.css
@@ -985,11 +985,11 @@
 
 # Install a RC file for the default GTK+ theme, and key themes
 install-data-local: install-ms-lib install-def-file install-mac-key-theme
-	$(mkdir_p) $(DESTDIR)$(datadir)/themes/Raleigh/gtk-3.0
+	$(MKDIR_P) $(DESTDIR)$(datadir)/themes/Raleigh/gtk-3.0
 	$(INSTALL_DATA) $(srcdir)/gtk.css.raleigh $(DESTDIR)$(datadir)/themes/Raleigh/gtk-3.0/gtk.css
-	$(mkdir_p) $(DESTDIR)$(datadir)/themes/Default/gtk-3.0
+	$(MKDIR_P) $(DESTDIR)$(datadir)/themes/Default/gtk-3.0
 	$(INSTALL_DATA) $(srcdir)/gtk-keys.css.default $(DESTDIR)$(datadir)/themes/Default/gtk-3.0/gtk-keys.css
-	$(mkdir_p) $(DESTDIR)$(datadir)/themes/Emacs/gtk-3.0
+	$(MKDIR_P) $(DESTDIR)$(datadir)/themes/Emacs/gtk-3.0
 	$(INSTALL_DATA) $(srcdir)/gtk-keys.css.emacs $(DESTDIR)$(datadir)/themes/Emacs/gtk-3.0/gtk-keys.css
 
 uninstall-local: uninstall-ms-lib uninstall-def-file uninstall-mac-key-theme
