Index: llvm-2.8/lib/ExecutionEngine/JIT/Intercept.cpp
===================================================================
--- llvm-2.8.orig/lib/ExecutionEngine/JIT/Intercept.cpp	2012-05-14 18:31:21.008318473 +0200
+++ llvm-2.8/lib/ExecutionEngine/JIT/Intercept.cpp	2012-05-14 18:32:53.523734850 +0200
@@ -19,6 +19,7 @@
 #include "llvm/Support/ErrorHandling.h"
 #include "llvm/System/DynamicLibrary.h"
 #include "llvm/Config/config.h"
+#include <unistd.h>
 using namespace llvm;
 
 // AtExitHandlers - List of functions to call when the program exits,
