Index: gcc-4.5.0/libstdc++-v3/configure
===================================================================
--- gcc-4.5.0.orig/libstdc++-v3/configure	2010-06-25 10:48:37.488384191 -0700
+++ gcc-4.5.0/libstdc++-v3/configure	2010-06-25 10:51:23.804380413 -0700
@@ -18262,6 +18262,9 @@
     cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <complex.h>
+#ifdef __UCLIBC__
+#error ugly hack to make sure configure test fails here for cross until uClibc supports the complex funcs
+#endif
 int
 main ()
 {
