Imported from Fedora

Index: M2Crypto-0.30.1/SWIG/_m2crypto.i
===================================================================
--- M2Crypto-0.30.1.orig/SWIG/_m2crypto.i
+++ M2Crypto-0.30.1/SWIG/_m2crypto.i
@@ -7,7 +7,7 @@
  * Copyright (c) 2009-2010 Heikki Toivonen. All rights reserved.
  *
  */
-
+%import "gcc_macros.h"
 %module(threads=1) m2crypto
 /* We really don't need threadblock (PyGILState_Ensure() etc.) anywhere.
    Disable threadallow as well, only enable it for operations likely to
@@ -15,11 +15,6 @@
 %nothreadblock;
 %nothreadallow;
 
-#if SWIG_VERSION >= 0x030000
-#define __WCHAR_MAX__ __WCHAR_MAX
-#define __WCHAR_MIN__ __WCHAR_MIN
-#endif
-
 %{
 #ifdef _WIN32
 #define _WINSOCKAPI_
