Fix dialog support

Upstream-Status: Pending

diff --git a/openbox/client.c b/openbox/client.c
index 4d54b38..c65e350 100644
--- a/openbox/client.c
+++ b/openbox/client.c
@@ -562,6 +562,14 @@ void client_manage(Window window, ObPrompt *prompt)
     /* free the ObAppSettings shallow copy */
     g_free(settings);
 
+    /* force dialogs to skip taskbar */
+    if (self->type == OB_CLIENT_TYPE_DIALOG) {
+        self->skip_taskbar = TRUE;
+
+        gulong state = prop_atoms.net_wm_state_skip_taskbar;
+        PROP_SET32(self->window, net_wm_state, atom, state);
+    }
+
     ob_debug("Managed window 0x%lx plate 0x%x (%s)\n",
              window, self->frame->window, self->class);
 
@@ -1733,6 +1741,14 @@ void client_setup_decor_and_functions(ObClient *self, gboolean reconfig)
         /* sometimes apps make dialog windows fullscreen for some reason (for
            e.g. kpdf does this..) */
         self->functions |= OB_CLIENT_FUNC_FULLSCREEN;
+        /* these windows cannot be maximized or minimized*/
+        self->decorations &= ~(OB_FRAME_DECOR_ICONIFY |
+                               OB_FRAME_DECOR_MAXIMIZE |
+                               OB_FRAME_DECOR_HANDLE);
+
+        self->functions &= ~(OB_CLIENT_FUNC_ICONIFY |
+                             OB_CLIENT_FUNC_MAXIMIZE |
+                             OB_CLIENT_FUNC_RESIZE);
         break;
 
     case OB_CLIENT_TYPE_UTILITY:
