--- mysql-4.1.22/configure.in.old	2008-09-18 20:11:15.000000000 -0500
+++ mysql-4.1.22/configure.in	2008-09-18 20:12:28.000000000 -0500
@@ -456,9 +456,9 @@ else
   fi
 fi
 
-AC_SUBST(HOSTNAME)
-AC_SUBST(PERL)
-AC_SUBST(PERL5)
+AC_SUBST(HOSTNAME,/bin/hostname)
+AC_SUBST(PERL,$(bindir)/perl)
+AC_SUBST(PERL5,$(bindir)/perl)
 
 # for build ndb docs
 
@@ -516,16 +516,17 @@ AC_MSG_RESULT("$FIND_PROC")
 
 # Check if a pid is valid
 AC_PATH_PROG(KILL, kill, kill)
+AC_SUBST(KILL,/bin/kill)
 AC_MSG_CHECKING("for kill switches")
-if $ac_cv_path_KILL -0 $$
+if $KILL -0 $$
 then
-  CHECK_PID="$ac_cv_path_KILL -0 \$\$PID > /dev/null 2> /dev/null"
+  CHECK_PID="$KILL -0 \$\$PID > /dev/null 2> /dev/null"
 elif kill -s 0 $$
 then
-  CHECK_PID="$ac_cv_path_KILL -s 0 \$\$PID > /dev/null 2> /dev/null"
+  CHECK_PID="$KILL -s 0 \$\$PID > /dev/null 2> /dev/null"
 else
   AC_MSG_WARN([kill -0 to check for pid seems to fail])
-    CHECK_PID="$ac_cv_path_KILL -s SIGCONT \$\$PID > /dev/null 2> /dev/null"
+    CHECK_PID="$KILL -s SIGCONT \$\$PID > /dev/null 2> /dev/null"
 fi
 AC_SUBST(CHECK_PID)
 AC_MSG_RESULT("$CHECK_PID")
