From 8229218dff1105e9fb2bb2c7510910a0db98f3ef Mon Sep 17 00:00:00 2001
From: chros <chros@chrosGX620>
Date: Wed, 2 Aug 2017 12:48:27 +0100
Subject: [PATCH] Modfiy gcc v6.x fix for empty CXXFLAGS (See #10)

---
 configure.ac            |  1 +
 scripts/rak_compiler.m4 | 21 +++++++++++++++++----
 2 files changed, 18 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 2f29e3f9..a6df6b80 100644
--- a/configure.ac
+++ b/configure.ac
@@ -28,6 +28,7 @@ AC_C_BIGENDIAN(
     AC_MSG_ERROR([Could not determine endianness])
 )
 
+RAK_CHECK_CFLAGS
 RAK_CHECK_CXXFLAGS
 RAK_ENABLE_DEBUG
 RAK_ENABLE_EXTRA_DEBUG
diff --git a/scripts/rak_compiler.m4 b/scripts/rak_compiler.m4
index 87871abf..9a361bed 100644
--- a/scripts/rak_compiler.m4
+++ b/scripts/rak_compiler.m4
@@ -1,12 +1,25 @@
+AC_DEFUN([RAK_CHECK_CFLAGS], [
+
+  AC_MSG_CHECKING([for user-defined CFLAGS])
+
+  if test "$CFLAGS" = ""; then
+    unset CFLAGS
+    AC_MSG_RESULT([undefined])
+  else
+    AC_MSG_RESULT([user-defined "$CFLAGS"])
+  fi
+])
+
+
 AC_DEFUN([RAK_CHECK_CXXFLAGS], [
 
   AC_MSG_CHECKING([for user-defined CXXFLAGS])
 
-  if test -n "$CXXFLAGS"; then
-    AC_MSG_RESULT([user-defined "$CXXFLAGS"])
+  if test "$CXXFLAGS" = ""; then
+    unset CXXFLAGS
+    AC_MSG_RESULT([undefined])
   else
-    CXXFLAGS="-O2"
-    AC_MSG_RESULT([default "$CXXFLAGS"])
+    AC_MSG_RESULT([user-defined "$CXXFLAGS"])
   fi
 ])
 
