From cbd946b6cad8c93b3d39ab4f338b3640f684cbfc Mon Sep 17 00:00:00 2001
From: Adam Fontenot <adam.m.fontenot@gmail.com>
Date: Sat, 7 Jul 2018 16:52:07 -0700
Subject: [PATCH] Fixes https://github.com/rakshasa/rtorrent/issues/731

---
 src/tracker/tracker_http.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/tracker/tracker_http.cc b/src/tracker/tracker_http.cc
index eefe5a17..1bf94107 100644
--- a/src/tracker/tracker_http.cc
+++ b/src/tracker/tracker_http.cc
@@ -145,13 +145,13 @@ TrackerHttp::send_state(int state) {
   if (!localAddress->is_address_any())
     s << "&ip=" << localAddress->address_str();
   
-  if (localAddress->is_address_any() || localAddress->family() != rak::socket_address::pf_inet6) {
+  if (localAddress->is_address_any() && localAddress->family() == rak::socket_address::pf_inet) {
     rak::socket_address local_v6;
     if (get_local_address(rak::socket_address::af_inet6, &local_v6))
       s << "&ipv6=" << rak::copy_escape_html(local_v6.address_str());
   }
 
-  if (localAddress->is_address_any() || localAddress->family() != rak::socket_address::pf_inet) {
+  if (localAddress->is_address_any() && localAddress->family() == rak::socket_address::pf_inet6) {
     rak::socket_address local_v4;
     if (get_local_address(rak::socket_address::af_inet, &local_v4))
       s << "&ipv4=" << local_v4.address_str();
