From d49316117401cff4045b1d1194cec60909e45555 Mon Sep 17 00:00:00 2001
From: chros <chros@chrosGX620>
Date: Sun, 13 May 2018 11:34:33 +0100
Subject: [PATCH] Fix honoring throttle.min_peers* settings in rtorrent (See
 #13)

---
 src/download/download_main.cc | 17 ++++++-----------
 1 file changed, 6 insertions(+), 11 deletions(-)

diff --git a/src/download/download_main.cc b/src/download/download_main.cc
index 48222c38..efe91d66 100644
--- a/src/download/download_main.cc
+++ b/src/download/download_main.cc
@@ -355,19 +355,14 @@ DownloadMain::receive_tracker_success() {
 
 void
 DownloadMain::receive_tracker_request() {
-  bool should_stop = false;
-  bool should_start = false;
+  if (info()->is_pex_enabled() && info()->size_pex() > 0
+      || connection_list()->size() + peer_list()->available_list()->size() / 2 >= connection_list()->min_size()) {
 
-  if (info()->is_pex_enabled() && info()->size_pex() > 0)
-    should_stop = true;
-
-  if (connection_list()->size() + peer_list()->available_list()->size() / 2 < connection_list()->min_size())
-    should_start = true;
-
-  if (should_stop)
     m_tracker_controller->stop_requesting();
-  else if (should_start)
-    m_tracker_controller->start_requesting();
+    return;
+  }
+
+  m_tracker_controller->start_requesting();
 }
 
 struct SocketAddressCompact_less {
