From 9fd2f35f86397b9259e0c9b7c54444c0ec3c8965 Mon Sep 17 00:00:00 2001
From: adam <adam@shinka.sh>
Date: Mon, 26 Feb 2018 16:04:34 +0200
Subject: [PATCH] increase piece length max

---
 src/download/download_constructor.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/download/download_constructor.cc b/src/download/download_constructor.cc
index 1bb261bd..67ef4276 100644
--- a/src/download/download_constructor.cc
+++ b/src/download/download_constructor.cc
@@ -157,7 +157,7 @@ DownloadConstructor::parse_info(const Object& b) {
   } else {
     chunkSize = b.get_key_value("piece length");
 
-    if (chunkSize <= (1 << 10) || chunkSize > (128 << 20))
+    if (chunkSize <= (1 << 10) || chunkSize > (128 << 22))
       throw input_error("Torrent has an invalid \"piece length\".");
   }
 
