From b263b347f14b9d0ca54d04ca92367a98d791dc17 Mon Sep 17 00:00:00 2001
From: Jari Sundell <sundell.software@gmail.com>
Date: Sun, 9 Jun 2019 14:53:32 +0900
Subject: [PATCH] Set max piece size 512mb.

---
 src/download/download_constructor.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/download/download_constructor.cc b/src/download/download_constructor.cc
index 67ef4276..1bf362fa 100644
--- a/src/download/download_constructor.cc
+++ b/src/download/download_constructor.cc
@@ -157,7 +157,7 @@ DownloadConstructor::parse_info(const Object& b) {
   } else {
     chunkSize = b.get_key_value("piece length");
 
-    if (chunkSize <= (1 << 10) || chunkSize > (128 << 22))
+    if (chunkSize <= (1 << 10) || chunkSize > (512 << 20))
       throw input_error("Torrent has an invalid \"piece length\".");
   }
 
