From cdfb4381bedb278dabb8ca75c10858d16b895355 Mon Sep 17 00:00:00 2001
From: Jari Sundell <sundell.software@gmail.com>
Date: Sun, 9 Jun 2019 14:55:44 +0900
Subject: [PATCH] Fixed compiler warning.

---
 src/download/download_main.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/download/download_main.cc b/src/download/download_main.cc
index efe91d66..9a3f9df2 100644
--- a/src/download/download_main.cc
+++ b/src/download/download_main.cc
@@ -355,7 +355,7 @@ DownloadMain::receive_tracker_success() {
 
 void
 DownloadMain::receive_tracker_request() {
-  if (info()->is_pex_enabled() && info()->size_pex() > 0
+  if ((info()->is_pex_enabled() && info()->size_pex()) > 0
       || connection_list()->size() + peer_list()->available_list()->size() / 2 >= connection_list()->min_size()) {
 
     m_tracker_controller->stop_requesting();
