From e813c344b1e4aa89288febb2f59109972083f1bb Mon Sep 17 00:00:00 2001
From: Rosen Penev <rosenp@gmail.com>
Date: Thu, 5 Dec 2019 01:55:53 -0800
Subject: [PATCH] Fix log format so GCC can check it. (#205)

---
 src/torrent/utils/log.cc | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/torrent/utils/log.cc b/src/torrent/utils/log.cc
index b855a2c6..24932996 100644
--- a/src/torrent/utils/log.cc
+++ b/src/torrent/utils/log.cc
@@ -16,6 +16,8 @@
 #include <functional>
 #include <memory>
 
+#define GROUPFMT (group >= LOG_NON_CASCADING) ? ("%" PRIi32 " ") : ("%" PRIi32 " %c ")
+
 namespace torrent {
 
 struct log_cache_entry {
@@ -356,9 +358,7 @@ log_gz_file_write(std::shared_ptr<log_gz_output>& outfile, const char* data, siz
 
   // Normal groups are nul-terminated strings.
   if (group >= 0) {
-    const char* fmt = (group >= LOG_NON_CASCADING) ? ("%" PRIi32 " ") : ("%" PRIi32 " %c ");
-
-    int buffer_length = snprintf(buffer, 64, fmt,
+    int buffer_length = snprintf(buffer, 64, GROUPFMT,
                                  cachedTime.seconds(),
                                  log_level_char[group % 6]);
     
