Index: gpsd-2.95/Makefile.am
===================================================================
--- gpsd-2.95.orig/Makefile.am	2010-08-26 15:31:38.804330723 +0200
+++ gpsd-2.95/Makefile.am	2010-08-26 15:31:57.170020341 +0200
@@ -253,9 +253,9 @@
 # Warning: This overrides autoconf's normal link-line generation process
 if LIBGPSMM_ENABLE
 libgps_la_SOURCES += libgpsmm.cpp
-libgps_la_LINK = /bin/sh ./libtool --tag=CXX --mode=link g++ $(libgps_la_LDFLAGS) -o $@
+libgps_la_LINK = $(LIBTOOL) --tag=CXX --mode=link g++ $(libgps_la_LDFLAGS) -o $@
 else
-libgps_la_LINK = /bin/sh ./libtool --tag=CC --mode=link gcc $(libgps_la_LDFLAGS) -o $@
+libgps_la_LINK = $(LIBTOOL) --tag=CC --mode=link gcc $(libgps_la_LDFLAGS) -o $@
 endif
 
 nodist_libgpsd_la_SOURCES = packet_names.h ais_json.i
