---
 configure.in |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

Index: DirectFB-1.2.8/configure.in
===================================================================
--- DirectFB-1.2.8.orig/configure.in
+++ DirectFB-1.2.8/configure.in
@@ -1288,8 +1288,11 @@ fi
 
 enable_tslib=no
 if test "$checkfor_tslib" = "yes"; then
-  PKG_CHECK_MODULES([TSLIB], [tslib-0.0], [enable_tslib=yes], [enable_tslib=no,
-    AC_MSG_WARN([*** no tslib -- tslib driver will not be built.])])
+  PKG_CHECK_MODULES([TSLIB], [tslib-1.0 >= 1.0.0], [enable_tslib=yes], [enable_tslib=no])
+  if test "$enable_tslib" = "no"; then
+    PKG_CHECK_MODULES([TSLIB], [tslib-0.0], [enable_tslib=yes], [enable_tslib=no,
+      AC_MSG_WARN([*** no tslib -- tslib driver will not be built.])])
+  fi
 fi
 
 enable_ucb1x00_ts=no
