Index: xserver-common-1.34/X11/xserver-common
===================================================================
--- xserver-common-1.34.orig/X11/xserver-common	2011-10-10 22:36:14.163129002 +0200
+++ xserver-common-1.34/X11/xserver-common	2011-10-10 22:37:47.513129001 +0200
@@ -129,6 +129,9 @@
         "herring")
             DPI="235"
             ARGS="$ARGS -dpi ${DPI} -nocursor";;
+        "mahimahi")
+            DPI="254"
+            ARGS="$ARGS -dpi ${DPI}";;
         "generic_omap1510/1610/1710")
             ARGS="$ARGS -screen ${SCREEN_SIZE}"
             DPI="225" ;;
