Index: lss/linux_syscall_support.h
===================================================================
--- lss.orig/linux_syscall_support.h
+++ lss/linux_syscall_support.h
@@ -118,15 +118,7 @@ extern "C" {
 #include <endian.h>
 
 #ifdef __mips__
-/* Include definitions of the ABI currently in use.                          */
-#ifdef __ANDROID__
-/* Android doesn't have sgidefs.h, but does have asm/sgidefs.h,
- * which has the definitions we need.
- */
 #include <asm/sgidefs.h>
-#else
-#include <sgidefs.h>
-#endif
 #endif
 #endif
 
