From 5039e7834fff94f8393963ea8dbad81fa95ef810 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@gmx.de>
Date: Fri, 16 Sep 2011 21:09:05 +0200
Subject: [PATCH] docs: On installation create relative paths in symlinks to
 images
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit


Signed-off-by: Andreas Müller <schnitzeltony@gmx.de>
---
 doc/manual/Makefile.am |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/doc/manual/Makefile.am b/doc/manual/Makefile.am
index 646821d..cbf79e1 100644
--- a/doc/manual/Makefile.am
+++ b/doc/manual/Makefile.am
@@ -20,7 +20,7 @@ install-data-local:
 			done; \
 			if test x"$$lang" != x"C" -a ! -d $(srcdir)/images/$$lang; then \
 				echo "--- Creating symlink to C images for $$lang"; \
-				ln -sf $(docdir)/html/C/images $(DESTDIR)$(docdir)/html/$$lang/images; \
+				ln -sf ../C/images $(DESTDIR)$(docdir)/html/$$lang/images; \
 			fi \
 		fi \
 	done;
-- 
1.7.4.4

