From 2e36a24997feb30ea8325eca5575e2e968909a5f Mon Sep 17 00:00:00 2001
From: Jason Kridner <jdk@ti.com>
Date: Thu, 20 Oct 2011 14:45:34 -0400
Subject: [PATCH] remove x86-specific compiler flags

---
 wscript |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/../o3/wscript b/../o3/wscript
index 6e21bee..607e1c3 100644
--- a/../o3/wscript
+++ b/../o3/wscript
@@ -9,8 +9,6 @@ def configure(conf):
   conf.check_tool('compiler_cxx')
   conf.check_tool('node_addon')   
   conf.add_os_flags('LDFLAGS','LINKFLAGS')
-  conf.env.append_value('CCFLAGS', ['-O3', '-msse2', '-ffast-math'])
-  conf.env.append_value('CXXFLAGS', ['-O3', '-msse2', '-ffast-math'])
 
 def build(bld):
   obj = bld.new_task_gen('cxx', 'shlib', 'node_addon')  
-- 
1.7.4.1

