Index: DirectFB-1.1.1/directfb-internal.pc.in
===================================================================
--- DirectFB-1.1.1.orig/directfb-internal.pc.in	2007-08-07 21:43:00.000000000 +0200
+++ DirectFB-1.1.1/directfb-internal.pc.in	2008-07-17 21:00:47.424654304 +0200
@@ -2,10 +2,10 @@
 exec_prefix=@exec_prefix@
 moduledir=@MODULEDIR@
 moduledirname=@MODULEDIRNAME@
-includedir=@INTERNALINCLUDEDIR@
+includedir=@includedir@
 
 Name: DirectFB-Internal
 Description: Third party module support package for DirectFB.
 Version: @VERSION@
 Requires: directfb = @VERSION@
-Cflags: @DFB_INTERNAL_CFLAGS@ -I@INTERNALINCLUDEDIR@
+Cflags: @DFB_INTERNAL_CFLAGS@ -I${includedir}/directfb -I${includedir}
Index: DirectFB-1.1.1/directfb.pc.in
===================================================================
--- DirectFB-1.1.1.orig/directfb.pc.in	2007-12-15 16:23:16.000000000 +0100
+++ DirectFB-1.1.1/directfb.pc.in	2008-07-17 20:59:58.044985193 +0200
@@ -9,4 +9,4 @@
 Requires: fusion direct
 Libs: -L${libdir} -ldirectfb @THREADLIB@ @OSX_LIBS@
 Libs.private: -L${libdir} @DYNLIB@ @ZLIB_LIBS@
-Cflags: @THREADFLAGS@ -I@INCLUDEDIR@
+Cflags: @THREADFLAGS@ -I${includedir}/directfb
Index: DirectFB-1.1.1/lib/fusion/fusion.pc.in
===================================================================
--- DirectFB-1.1.1.orig/lib/fusion/fusion.pc.in	2007-08-07 21:43:00.000000000 +0200
+++ DirectFB-1.1.1/lib/fusion/fusion.pc.in	2008-07-17 20:58:07.051338866 +0200
@@ -8,4 +8,4 @@
 Version: @VERSION@
 Requires: direct
 Libs: -L${libdir} -lfusion
-Cflags: -I@INCLUDEDIR@
+Cflags: -I${includedir}/directfb -I${includedir}
Index: DirectFB-1.1.1/lib/voodoo/voodoo.pc.in
===================================================================
--- DirectFB-1.1.1.orig/lib/voodoo/voodoo.pc.in	2007-08-07 21:43:00.000000000 +0200
+++ DirectFB-1.1.1/lib/voodoo/voodoo.pc.in	2008-07-17 20:58:07.054671842 +0200
@@ -8,4 +8,4 @@
 Version: @VERSION@
 Requires: direct
 Libs: -L${libdir} -lvoodoo
-Cflags: -I@INCLUDEDIR@
+Cflags: -I${includedir}/directfb -I${includedir}
