Index: DirectFB-1.1.1/configure.in
===================================================================
--- DirectFB-1.1.1.orig/configure.in	2008-05-07 23:08:04.000000000 +0200
+++ DirectFB-1.1.1/configure.in	2008-05-07 23:15:43.000000000 +0200
@@ -1292,8 +1292,11 @@
 
 enable_tslib=no
 if test "$checkfor_tslib" = "yes"; then
+  PKG_CHECK_MODULES([TSLIB], [tslib-1.0 >= 1.0.0], [enable_tslib=yes], [enable_tslib=no])
+  if test "$enable_tslib" = "no"; then
     PKG_CHECK_MODULES([TSLIB], [tslib-0.0 >= 0.0.2], enable_tslib=yes, enable_tslib=no
         AC_MSG_WARN([*** no tslib -- tslib driver will not be built.]))
+  fi
 fi
 
 enable_ucb1x00_ts=no
