tests: also install tests app

Upstream-Status: Pending

Signed-off-by: Yu Ke <ke.yu@intel.com>

diff --git a/tests/Makefile.am b/tests/Makefile.am
index bf1987f..d909e19 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -6,10 +6,11 @@ AM_CPPFLAGS = \
 
 LDADD = $(top_builddir)/libdrm.la
 
-check_PROGRAMS = \
+bin_PROGRAMS = \
 	dristat \
 	drmstat
 
+check_PROGRAMS =
 SUBDIRS = modeprint
 
 if HAVE_LIBKMS
diff --git a/tests/modeprint/Makefile.am b/tests/modeprint/Makefile.am
index c4862ac..7db76ea 100644
--- a/tests/modeprint/Makefile.am
+++ b/tests/modeprint/Makefile.am
@@ -2,7 +2,7 @@ AM_CFLAGS = \
 	-I$(top_srcdir)/include/drm \
 	-I$(top_srcdir)
 
-noinst_PROGRAMS = \
+bin_PROGRAMS = \
 	modeprint
 
 modeprint_SOURCES = \
diff --git a/tests/modetest/Makefile.am b/tests/modetest/Makefile.am
index 2191242..1d2c1b0 100644
--- a/tests/modetest/Makefile.am
+++ b/tests/modetest/Makefile.am
@@ -4,7 +4,7 @@ AM_CFLAGS = \
 	-I$(top_srcdir) \
 	$(CAIRO_CFLAGS)
 
-noinst_PROGRAMS = \
+bin_PROGRAMS = \
 	modetest
 
 modetest_SOURCES = \
