
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- udev-062/udev.c~noasmlinkage.patch
+++ udev-062/udev.c
@@ -54,7 +54,7 @@
 }
 #endif
 
-static void asmlinkage sig_handler(int signum)
+static void sig_handler(int signum)
 {
 	switch (signum) {
 		case SIGALRM:
--- udev-062/udevd.c~noasmlinkage.patch
+++ udev-062/udevd.c
@@ -639,7 +639,7 @@
 	return msg;
 }
 
-static void asmlinkage sig_handler(int signum)
+static void sig_handler(int signum)
 {
 	int rc;
 
--- udev-062/udevstart.c~noasmlinkage.patch
+++ udev-062/udevstart.c
@@ -323,7 +323,7 @@
 	exec_list(&device_list);
 }
 
-static void asmlinkage sig_handler(int signum)
+static void sig_handler(int signum)
 {
 	switch (signum) {
 		case SIGALRM:
