From 953cff93c85d3cfd5cbcac56e14443dc5f6e5fbd Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 19 Oct 2016 01:57:16 +0000
Subject: [PATCH] ignore constant-logical-operand warning with clang

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/uefi/uefirtauthvar/uefirtauthvar.c | 5 +++++
 1 file changed, 5 insertions(+)

Index: git/src/uefi/uefirtauthvar/uefirtauthvar.c
===================================================================
--- git.orig/src/uefi/uefirtauthvar/uefirtauthvar.c
+++ git/src/uefi/uefirtauthvar/uefirtauthvar.c
@@ -142,6 +142,9 @@ static int uefirtauthvar_deinit(fwts_fra
 	return FWTS_OK;
 }
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wconstant-logical-operand"
+
 static int check_fw_support(fwts_framework *fw, uint64_t status)
 {
 	if ((status == EFI_INVALID_PARAMETER) &&
@@ -172,6 +175,8 @@ static int check_fw_support(fwts_framewo
 	return FWTS_OK;
 }
 
+#pragma clang diagnostic pop
+
 /*
  * Set the created authenticated variable, AuthVarCreate,
  * and checking the data size and data.
Index: git/src/lib/src/fwts_acpi_tables.c
===================================================================
--- git.orig/src/lib/src/fwts_acpi_tables.c
+++ git/src/lib/src/fwts_acpi_tables.c
@@ -392,10 +392,14 @@ static int fwts_acpi_handle_fadt(
 	/* Determine FACS addr and load it.
 	 * Will ignore the missing FACS in the hardware-reduced mode.
 	 */
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Waddress-of-packed-member"
+
 	result = fwts_acpi_handle_fadt_tables(fw, fadt,
 			"FACS", "FIRMWARE_CTRL", "X_FIRMWARE_CTRL",
 			&fadt->firmware_control, &fadt->x_firmware_ctrl,
 			provenance);
+
 	if (result != FWTS_OK) {
 		if ((result == FWTS_NULL_POINTER) &&
 				fwts_acpi_is_reduced_hardware(fadt)) {
@@ -414,6 +418,7 @@ static int fwts_acpi_handle_fadt(
 		return FWTS_ERROR;
 	}
 	return FWTS_OK;
+#pragma clang diagnostic pop
 }
 
 /*
Index: git/src/dmi/dmicheck/dmicheck.c
===================================================================
--- git.orig/src/dmi/dmicheck/dmicheck.c
+++ git/src/dmi/dmicheck/dmicheck.c
@@ -209,6 +209,8 @@ static const char *uuid_patterns[] = {
 	"0A0A0A0A-0A0A-0A0A-0A0A-0A0A0A0A0A0A",
 	NULL,
 };
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunneeded-internal-declaration"
 
 static const fwts_chassis_type_map fwts_dmi_chassis_type[] = {
 	{ "Invalid",		FWTS_SMBIOS_CHASSIS_INVALID },
@@ -245,6 +247,7 @@ static const fwts_chassis_type_map fwts_
 	{ "Convertible",	FWTS_SMBIOS_CHASSIS_CONVERTIBLE },
 	{ "Detachable",		FWTS_SMBIOS_CHASSIS_DETACHABLE },
 };
+#pragma clang diagnostic pop
 
 /* Remapping table from buggy version numbers to correct values */
 static const fwts_dmi_version dmi_versions[] = {
