Make the Makefile obey LDFLAGS.

Upstream-Status: Pending

--- a/src/Makefile.in	2006-04-02 18:29:42.000000000 +0400
+++ b/src/Makefile.in	2010-04-16 17:51:31.000000000 +0400
@@ -67,16 +67,16 @@ all: $(TARGETS)
 	@echo "Type 'make install' as root to install the software."
 
 pppoe-sniff: pppoe-sniff.o if.o common.o debug.o
-	@CC@ -o pppoe-sniff pppoe-sniff.o if.o common.o debug.o
+	@CC@ -o pppoe-sniff pppoe-sniff.o if.o common.o debug.o $(LDFLAGS)
 
 pppoe-server: pppoe-server.o if.o debug.o common.o md5.o libevent/libevent.a @PPPOE_SERVER_DEPS@
-	@CC@ -o pppoe-server @RDYNAMIC@ pppoe-server.o if.o debug.o common.o md5.o $(PPPOE_SERVER_LIBS) -Llibevent -levent
+	@CC@ -o pppoe-server @RDYNAMIC@ pppoe-server.o if.o debug.o common.o md5.o $(PPPOE_SERVER_LIBS) -Llibevent -levent $(LDFLAGS)
 
 pppoe: pppoe.o if.o debug.o common.o ppp.o discovery.o
-	@CC@ -o pppoe pppoe.o if.o debug.o common.o ppp.o discovery.o
+	@CC@ -o pppoe pppoe.o if.o debug.o common.o ppp.o discovery.o $(LDFLAGS)
 
 pppoe-relay: relay.o if.o debug.o common.o
-	@CC@ -o pppoe-relay relay.o if.o debug.o common.o
+	@CC@ -o pppoe-relay relay.o if.o debug.o common.o $(LDFLAGS)
 
 pppoe.o: pppoe.c pppoe.h
 	@CC@ $(CFLAGS) '-DVERSION="$(VERSION)"' -c -o pppoe.o pppoe.c
