From 9ed4940f796fabd86ace513c67ea3208815c553b Mon Sep 17 00:00:00 2001
From: Robert Berger <Robert.Berger@ReliableEmbeddedSystems.com>
Date: Wed, 5 Mar 2025 12:02:38 +0000
Subject: [PATCH] add sparse support

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Robert Berger <Robert.Berger@ReliableEmbeddedSystems.com>
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/memory_driver/Makefile b/memory_driver/Makefile
index 8b9bdbd..ae7151b 100644
--- a/memory_driver/Makefile
+++ b/memory_driver/Makefile
@@ -7,8 +7,8 @@ all:
 
 # if you would like to invoke sparse:
 #
-# all:
-#       $(MAKE) KBUILD_CHECKSRC=1 -C $(KERNEL_SRC) M=$(SRC)
+all:
+	$(MAKE) KBUILD_CHECKSRC=1 -C $(KERNEL_SRC) M=$(SRC)
 
 modules_install:
 	$(MAKE) -C $(KERNEL_SRC) M=$(SRC) modules_install
-- 
2.34.1

