Index: dbus/tools/Makefile.am
===================================================================
--- dbus.orig/tools/Makefile.am
+++ dbus/tools/Makefile.am
@@ -16,7 +16,7 @@ endif
 bin_PROGRAMS = dbusxx-xml2cpp dbusxx-introspect
 
 dbusxx_xml2cpp_SOURCES = xml.h xml.cpp xml2cpp.h xml2cpp.cpp
-dbusxx_xml2cpp_LDADD = $(libdbus_cxx_la)
+dbusxx_xml2cpp_LDADD = $(libdbus_cxx_la) -lexpat
 
 dbusxx_introspect_SOURCES = introspect.h introspect.cpp
 dbusxx_introspect_LDADD = $(libdbus_cxx_la)
