From 7f0e0613abe8aa1861946e2cc25397690da7128b Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 31 Mar 2017 16:05:05 -0700
Subject: [PATCH 2/7] socket-util.h: Include linux/sockios.h on musl

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/systemd/src/basic/socket-util.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/systemd/src/basic/socket-util.h b/src/systemd/src/basic/socket-util.h
index d7e2d85..2e7f7b6 100644
--- a/src/systemd/src/basic/socket-util.h
+++ b/src/systemd/src/basic/socket-util.h
@@ -30,6 +30,12 @@
 #include <linux/if_infiniband.h>
 #include <linux/if_packet.h>
 
+#if !defined(__GLIBC__)
+/* SIOCGSTAMPNS from linux/asm-generic.h
+ * for src/systemd/src/libsystemd-network/sd-lldp.c */
+#include <linux/sockios.h>
+#endif
+
 #include "macro.h"
 #include "missing.h"
 #include "util.h"
-- 
2.14.1

