Index: git/Makefile
===================================================================
--- git.orig/Makefile	2014-07-17 20:53:23.620479788 +0000
+++ git/Makefile	2014-07-17 20:56:07.992484267 +0000
@@ -45,7 +45,7 @@
 # about it if it doesn't.
 LIBUSB = $(shell pkg-config --libs libusb-1.0 2> /dev/null)
 
-LIBXML2 = $(shell xml2-config --libs)
+LIBXML2 = $(shell pkg-config --libs libxml-2.0)
 LIBGTK = $(shell pkg-config --libs gtk+-2.0 glib-2.0 gconf-2.0)
 LIBDIVECOMPUTERCFLAGS = -I$(LIBDIVECOMPUTERINCLUDES)
 LIBDIVECOMPUTER = $(LIBDIVECOMPUTERARCHIVE) $(LIBUSB)
@@ -71,7 +71,7 @@
 	$(INSTALL) $(MANFILES) $(MANDIR)
 
 parse-xml.o: parse-xml.c dive.h
-	$(CC) $(CFLAGS) `pkg-config --cflags glib-2.0` -c `xml2-config --cflags`  parse-xml.c
+	$(CC) $(CFLAGS) `pkg-config --cflags glib-2.0` -c `pkg-config --cflags libxml-2.0`  parse-xml.c
 
 save-xml.o: save-xml.c dive.h
 	$(CC) $(CFLAGS) `pkg-config --cflags glib-2.0` -c save-xml.c
