From 1798dc93ab65be94e14faad33a19cab051eef788 Mon Sep 17 00:00:00 2001
From: yangxuenian <yangxn@cicvd.com>
Date: Thu, 11 Apr 2024 16:36:29 +0800
Subject: [PATCH] build shared lib and skip example

---
 CMakeLists.txt | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a72acb8..89abeda 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,7 +45,7 @@ else()
 endif()
 
 ## Add serial library
-add_library(${PROJECT_NAME} ${serial_SRCS})
+add_library(${PROJECT_NAME} SHARED ${serial_SRCS})
 if(APPLE)
 	target_link_libraries(${PROJECT_NAME} ${FOUNDATION_LIBRARY} ${IOKIT_LIBRARY})
 elseif(UNIX)
@@ -55,9 +55,15 @@ else()
 endif()
 
 ## Uncomment for example
-add_executable(serial_example examples/serial_example.cc)
-add_dependencies(serial_example ${PROJECT_NAME})
-target_link_libraries(serial_example ${PROJECT_NAME})
+#add_executable(serial_example examples/serial_example.cc)
+#add_dependencies(serial_example ${PROJECT_NAME})
+#target_link_libraries(serial_example ${PROJECT_NAME})
+
+set (SERIAL_VERSION_MAJOR 1)
+set (SERIAL_VERSION_MINOR 0)
+set (SERIAL_VERSION_PATCH 0)
+set (SERIAL_VERSION_STRING ${SERIAL_VERSION_MAJOR}.${SERIAL_VERSION_MINOR}.${SERIAL_VERSION_PATCH})
+set_target_properties (${PROJECT_NAME} PROPERTIES VERSION ${SERIAL_VERSION_STRING} SOVERSION ${SERIAL_VERSION_MAJOR})
 
 ## Include headers
 include_directories(include)
