From 2649787d76cfe7d89abb1b5afab7cb7cd54b601b Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Sun, 2 Oct 2011 19:54:29 +0200
Subject: [PATCH] docs: fix build without xsltproc

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---

Upstream-status: Pending

 Makefile.am  |    2 ++
 2 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index f4a17aa..01240a5 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -750,6 +750,7 @@ EXTRA_DIST += \
         src/dbus-loop.h \
         src/spawn-agent.h
 
+if HAVE_XSLTPROC
 MANPAGES = \
 	man/systemd.1 \
 	man/systemctl.1 \
@@ -844,6 +845,7 @@ EXTRA_DIST += \
 	$(XML_IN_FILES) \
 	${nodist_man_MANS:=.in} \
 	${XML_IN_FILES:.xml.in=.html.in}
+endif
 
 systemd_SOURCES = \
 	src/main.c
-- 
1.6.6.1

