From d228134ff5af3bd0ef1087e9bee66583c1546553 Mon Sep 17 00:00:00 2001
From: Dirk Eibach <eibach@gdsys.de>
Date: Wed, 29 Feb 2012 12:45:46 +0100
Subject: [PATCH] systemd-journald: fix endianess bug

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---

Upstream-status: Submitted

 src/journal/journal-file.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/journal/journal-file.c b/src/journal/journal-file.c
index 20ca3f6..275caea 100644
--- a/src/journal/journal-file.c
+++ b/src/journal/journal-file.c
@@ -238,7 +238,7 @@ static int journal_file_allocate(JournalFile *f, uint64_t offset, uint64_t size)
         if (fstat(f->fd, &f->last_stat) < 0)
                 return -errno;
 
-        f->header->arena_size = new_size - htole64(f->header->arena_offset);
+        f->header->arena_size = htole64(new_size - le64toh(f->header->arena_offset));
 
         return 0;
 }
-- 
1.7.7.4

