From afc9b136758f0d9b77367dac73a7415ca33ed091 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Mon, 27 Jun 2011 15:44:41 +0200
Subject: [PATCH] systemd: disable xml file stuff and introspection

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---
 Makefile.am |   38 ++------------------------------------
 1 files changed, 2 insertions(+), 36 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 15349a1..37a51aa 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -246,24 +246,6 @@ nodist_udevrules_DATA = \
         src/73-seat-late.rules \
         src/99-systemd.rules
 
-dbusinterface_DATA = \
-	org.freedesktop.systemd1.Manager.xml \
-	org.freedesktop.systemd1.Job.xml \
-	org.freedesktop.systemd1.Unit.xml \
-	org.freedesktop.systemd1.Service.xml \
-	org.freedesktop.systemd1.Socket.xml \
-	org.freedesktop.systemd1.Timer.xml \
-	org.freedesktop.systemd1.Target.xml \
-	org.freedesktop.systemd1.Device.xml \
-	org.freedesktop.systemd1.Mount.xml \
-	org.freedesktop.systemd1.Automount.xml \
-	org.freedesktop.systemd1.Snapshot.xml \
-	org.freedesktop.systemd1.Swap.xml \
-	org.freedesktop.systemd1.Path.xml \
-        org.freedesktop.hostname1.xml \
-        org.freedesktop.locale1.xml \
-        org.freedesktop.timedate1.xml
-
 dist_bashcompletion_DATA = \
 	src/systemctl-bash-completion.sh
 
@@ -739,17 +721,10 @@ XML_FILES = \
 XML_IN_FILES = \
 	${patsubst %.1,%.xml.in,${patsubst %.3,%.xml.in,${patsubst %.5,%.xml.in,${patsubst %.7,%.xml.in,${patsubst %.8,%.xml.in,$(nodist_man_MANS)}}}}}
 
-dist_noinst_DATA = \
-	${XML_FILES:.xml=.html}
-
-nodist_noinst_DATA = \
-	${XML_IN_FILES:.xml.in=.html}
-
 EXTRA_DIST += \
 	$(XML_FILES) \
 	$(XML_IN_FILES) \
-	${nodist_man_MANS:=.in} \
-	${XML_IN_FILES:.xml.in=.html.in}
+	${nodist_man_MANS:=.in}
 
 systemd_SOURCES = \
 	src/main.c
@@ -1504,7 +1479,6 @@ CLEANFILES = \
 	$(nodist_systemunit_DATA) \
 	$(nodist_userunit_DATA) \
 	$(nodist_man_MANS) \
-	${XML_IN_FILES:.xml.in=.html} \
 	$(pkgconfigdata_DATA) \
 	$(pkgconfiglib_DATA) \
 	$(nodist_polkitpolicy_DATA) \
@@ -1569,17 +1543,9 @@ man/%.8: man/%.xml
 man/%.8.in: man/%.xml.in
 	$(XSLTPROC_PROCESS_MAN_IN)
 
-man/%.html: man/%.xml
-	$(XSLTPROC_PROCESS_HTML)
-
-man/%.html.in: man/%.xml.in
-	$(XSLTPROC_PROCESS_HTML_IN)
-
 CLEANFILES += \
 	$(dist_man_MANS) \
-	${nodist_man_MANS:=.in} \
-	${XML_FILES:.xml=.html} \
-	${XML_IN_FILES:.xml.in=.html.in}
+	${nodist_man_MANS:=.in}
 endif
 
 DBUS_PREPROCESS = $(CPP) -P $(DBUS_CFLAGS) -imacros dbus/dbus-protocol.h
-- 
1.6.6.1

