From: Andrej Shadura <andrew.shadura@collabora.co.uk>
Date: Tue, 3 Jun 2025 14:50:29 +0200
Subject: Remove commented out lines resulting in incorrect shell syntax

Upstream-Status: Pending
Origin: Debian [https://salsa.debian.org/collabora-team/imx-code-signing-tool/-/blob/fb2961efc88b9b6f37e98a5feed26873713299ee/debian/patches/fix-shell-syntax.patch]
---
 src/tools/hab_srktool_scripts/createSRKTable | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/src/tools/hab_srktool_scripts/createSRKTable b/src/tools/hab_srktool_scripts/createSRKTable
index 171f430..85237df 100755
--- a/src/tools/hab_srktool_scripts/createSRKTable
+++ b/src/tools/hab_srktool_scripts/createSRKTable
@@ -164,10 +164,3 @@ printf "%04s" $SRKTableSize | \
 			dd of=SRK_table.bin bs=1 seek=1 conv=notrunc
 
 
-if [ $DEBUG = 0 ]; then
-
-#If hexdiff utility is installed you can compare SRK table generated
-#with SRK table from SRKTOOL as follows
-
-#hexdiff SRK_table.bin SRK_1_2_3_4_table.bin
-fi
