Description: 64 bit fix for libsmbclient
Author: Christian Perrier <bubulle@debian.org>
Bug-Debian: http://bugs.debian.org/221618
Forwarded: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=221618#27

Index: samba/source3/include/libsmbclient.h
===================================================================
--- samba.orig/source3/include/libsmbclient.h
+++ samba/source3/include/libsmbclient.h
@@ -79,6 +79,10 @@
 #include <fcntl.h>
 #include <utime.h>
 
+  /* Debian bug #221618 */
+#define _LARGEFILE64_SOURCE
+#define _FILE_OFFSET_BITS 64
+
 #define SMBC_BASE_FD        10000 /* smallest file descriptor returned */
 
 #define SMBC_WORKGROUP      1
