Index: gnome-settings-daemon-2.26.1/configure.ac
===================================================================
--- gnome-settings-daemon-2.26.1.orig/configure.ac	2009-09-16 22:57:31.000000000 +0100
+++ gnome-settings-daemon-2.26.1/configure.ac	2009-09-16 22:58:45.000000000 +0100
@@ -133,9 +133,6 @@
 #
 AC_DEFUN([AC_CHECK_X_HEADERS], [
   ac_save_CPPFLAGS="$CPPFLAGS"
-  if test \! -z "$includedir" ; then
-    CPPFLAGS="$CPPFLAGS -I$includedir"
-  fi
   CPPFLAGS="$CPPFLAGS $X_CFLAGS"
   AC_CHECK_HEADERS([$1],[$2],[$3],[$4])
   CPPFLAGS="$ac_save_CPPFLAGS"])
@@ -148,15 +145,9 @@
   ac_save_LDFLAGS="$LDFLAGS"
 #  ac_save_LIBS="$LIBS"
 
-  if test \! -z "$includedir" ; then
-    CPPFLAGS="$CPPFLAGS -I$includedir"
-  fi
   # note: $X_CFLAGS includes $x_includes
   CPPFLAGS="$CPPFLAGS $X_CFLAGS"
 
-  if test \! -z "$libdir" ; then
-    LDFLAGS="$LDFLAGS -L$libdir"
-  fi
   # note: $X_LIBS includes $x_libraries
   LDFLAGS="$LDFLAGS $ALL_X_LIBS"
 
