Index: mysql-5.1.40/config/ac-macros/misc.m4
===================================================================
--- mysql-5.1.40.orig/config/ac-macros/misc.m4
+++ mysql-5.1.40/config/ac-macros/misc.m4
@@ -476,7 +476,7 @@ AC_DEFUN([MYSQL_STACK_DIRECTION],
  {
    exit (find_stack_direction() < 0);
  }], ac_cv_c_stack_direction=1, ac_cv_c_stack_direction=-1,
-   ac_cv_c_stack_direction=)])
+   ac_cv_c_stack_direction=0)])
  AC_DEFINE_UNQUOTED(STACK_DIRECTION, $ac_cv_c_stack_direction)
 ])dnl
 
