Index: kexec-tools-2.0.2/purgatory/string.c
===================================================================
--- kexec-tools-2.0.2.orig/purgatory/string.c	2008-05-16 13:28:19.000000000 +0200
+++ kexec-tools-2.0.2/purgatory/string.c	2009-09-16 17:49:04.000000000 +0200
@@ -1,5 +1,5 @@
 #include <stddef.h>
-#include <string.h>
+/* #include <string.h> */

 size_t strnlen(const char *s, size_t max)
 {
