From 3d371334d5668bcd02a38ff99884bd343c244d68 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Sat, 18 Jun 2011 23:51:35 +0200
Subject: [PATCH 3/7] gatomic-proper-pointer-get-cast.patch

Upstream-Status: Unknown

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---
 glib/gatomic.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/glib/gatomic.h b/glib/gatomic.h
index ddd39b8..b758142 100644
--- a/glib/gatomic.h
+++ b/glib/gatomic.h
@@ -70,7 +70,7 @@ void     g_atomic_pointer_set                  (volatile gpointer G_GNUC_MAY_ALI
   (g_atomic_int_set) ((volatile gint G_GNUC_MAY_ALIAS *) (volatile void *) (atomic), (newval)))
 # define g_atomic_pointer_get(atomic) \
  ((void) sizeof (gchar [sizeof (*(atomic)) == sizeof (gpointer) ? 1 : -1]), \
-  (g_atomic_pointer_get) ((volatile gpointer G_GNUC_MAY_ALIAS *) (volatile void *) (atomic)))
+  (g_atomic_pointer_get) ((volatile gpointer G_GNUC_MAY_ALIAS *) (volatile void G_GNUC_MAY_ALIAS *) (atomic)))
 # define g_atomic_pointer_set(atomic, newval) \
  ((void) sizeof (gchar [sizeof (*(atomic)) == sizeof (gpointer) ? 1 : -1]), \
   (g_atomic_pointer_set) ((volatile gpointer G_GNUC_MAY_ALIAS *) (volatile void *) (atomic), (newval)))
-- 
1.6.6.1

