Index: claws-mail-2.9.1/claws-mail.desktop
===================================================================
--- claws-mail-2.9.1.orig/claws-mail.desktop	2007-04-24 17:40:20.000000000 +0100
+++ claws-mail-2.9.1/claws-mail.desktop	2007-04-25 07:08:36.000000000 +0100
@@ -1,11 +1,11 @@
 [Desktop Entry]
 Encoding=UTF-8
-Name=Claws Mail
+Name=Mail
 Exec=claws-mail
 Icon=claws-mail
-Info="Claws Mail"
+Info=Email Application
 Categories=GTK;Network;Email;
-Comment="Gtk+ based Mail Client"
+Comment=Email Application
 Terminal=false
 Type=Application
 StartupNotify=true
