From: Jari Aalto <jari.aalto@cante.net>
Subject: Correct fomat strings fro hardened build flags

---
Upstream-Status: Pending

 gpl/pngsplit.c |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/gpl/pngsplit.c b/gpl/pngsplit.c
index ca4567b..0147ef4 100644
--- a/gpl/pngsplit.c
+++ b/gpl/pngsplit.c
@@ -141,8 +141,8 @@ int main(int argc, char *argv[])
                   0 == strncmp( argv[argn], "-quiet", 2 ) )
             verbose = 0;
         else {
-            fprintf(stderr, PNGSPLIT_HDR);
-            fprintf(stderr, PNGSPLIT_USAGE);
+            fprintf(stderr, "%s", PNGSPLIT_HDR);
+            fprintf(stderr, "%s", PNGSPLIT_USAGE);
             fflush(stderr);
             return 1;
         }
@@ -150,8 +150,8 @@ int main(int argc, char *argv[])
     }
 
     if ( argn == argc ) {
-        fprintf(stderr, PNGSPLIT_HDR);
-        fprintf(stderr, PNGSPLIT_USAGE);
+        fprintf(stderr, "%s", PNGSPLIT_HDR);
+        fprintf(stderr, "%s", PNGSPLIT_USAGE);
         fflush(stderr);
         return 5;
     } else {
@@ -162,7 +162,7 @@ int main(int argc, char *argv[])
 
     /*=============================  MAIN LOOP  =============================*/
 
-    fprintf(stdout, PNGSPLIT_HDR);
+    fprintf(stdout, "%s", PNGSPLIT_HDR);
     fflush(stdout);
 
     while (argn <= argc) {
