Index: memcached-1.4.33/items.c
===================================================================
--- memcached-1.4.33.orig/items.c
+++ memcached-1.4.33/items.c
@@ -148,7 +148,7 @@ item *do_item_alloc(char *key, const siz
     uint8_t nsuffix;
     item *it = NULL;
     char suffix[40];
-    if (nbytes < 2 || nkey < 0)
+    if (nbytes < 2)
         return 0;
 
     size_t ntotal = item_make_header(nkey + 1, flags, nbytes, suffix, &nsuffix);
