From: Sergey Lapin <slapin@slind.org>
Date: Wed, 28 Jan 2009 16:34:15 +0300
Subject: [PATCH] minimal IEEE802.15.4 allowed

---
 tcpdump.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/tcpdump.c b/tcpdump.c
index 06683af..fbc944c 100644
--- a/tcpdump.c
+++ b/tcpdump.c
@@ -282,6 +282,9 @@ static struct printer printers[] = {
 #ifdef DLT_MFR
 	{ mfr_if_print,		DLT_MFR },
 #endif
+#ifdef DLT_IEEE802_15_4
+	{ raw_if_print, DLT_IEEE802_15_4 },
+#endif
 #if defined(DLT_BLUETOOTH_HCI_H4_WITH_PHDR) && defined(HAVE_PCAP_BLUETOOTH_H)
 	{ bt_if_print,		DLT_BLUETOOTH_HCI_H4_WITH_PHDR},
 #endif
