Upstream-Status: backport

Index: src/bin/edje_cc_out.c
===================================================================
--- a/src/bin/edje_cc_out.c	(revision 79483)
+++ b/src/bin/edje_cc_out.c	(revision 79484)
@@ -1900,27 +1900,32 @@
           {
              for (i = 0 ; i < pc->programs.fnmatch_count ; i++)
                {
-                  if (!strcmp(pl->u.ep->name, pc->programs.fnmatch[i]->name))
+                  if (pl->u.ep->name && pc->programs.fnmatch[i]->name &&
+                      !strcmp(pl->u.ep->name, pc->programs.fnmatch[i]->name))
                     data_queue_anonymous_lookup(pc, pc->programs.fnmatch[i], dest);
                }
              for (i = 0 ; i < pc->programs.strcmp_count ; i++)
                {
-                  if (!strcmp(pl->u.ep->name, pc->programs.strcmp[i]->name))
+                  if (pl->u.ep->name && pc->programs.strcmp[i]->name &&
+                      !strcmp(pl->u.ep->name, pc->programs.strcmp[i]->name))
                     data_queue_anonymous_lookup(pc, pc->programs.strcmp[i], dest);
                }
              for (i = 0 ; i < pc->programs.strncmp_count ; i++)
                {
-                  if (!strcmp(pl->u.ep->name, pc->programs.strncmp[i]->name))
+                  if (pl->u.ep->name && pc->programs.strncmp[i]->name &&
+                      !strcmp(pl->u.ep->name, pc->programs.strncmp[i]->name))
                     data_queue_anonymous_lookup(pc, pc->programs.strncmp[i], dest);
                }
              for (i = 0 ; i < pc->programs.strrncmp_count ; i++)
                {
-                  if (!strcmp(pl->u.ep->name, pc->programs.strrncmp[i]->name))
+                  if (pl->u.ep->name && pc->programs.strrncmp[i]->name &&
+                      !strcmp(pl->u.ep->name, pc->programs.strrncmp[i]->name))
                     data_queue_anonymous_lookup(pc, pc->programs.strrncmp[i], dest);
                }
              for (i = 0 ; i < pc->programs.nocmp_count ; i++)
                {
-                  if (!strcmp(pl->u.ep->name, pc->programs.nocmp[i]->name))
+                  if (pl->u.ep->name && pc->programs.nocmp[i]->name &&
+                      !strcmp(pl->u.ep->name, pc->programs.nocmp[i]->name))
                     data_queue_anonymous_lookup(pc, pc->programs.nocmp[i], dest);
                }
           }
@@ -2466,6 +2471,7 @@
 _data_queue_part_lookup(Edje_Part_Collection *pc, char *name, char *ptr, int len)
 {
    Code_Lookup *cl;
+
    cl = mem_alloc(SZ(Code_Lookup));
    cl->ptr = ptr;
    cl->len = len;
