Index: mysql-5.1.40/storage/innodb_plugin/plug.in
===================================================================
--- mysql-5.1.40.orig/storage/innodb_plugin/plug.in
+++ mysql-5.1.40/storage/innodb_plugin/plug.in
@@ -63,35 +63,6 @@ MYSQL_PLUGIN_ACTIONS(innodb_plugin,  [
 		;;
   esac
   AC_SUBST(INNODB_DYNAMIC_CFLAGS)
-  AC_MSG_CHECKING(whether pthread_t can be used by GCC atomic builtins)
-  AC_TRY_RUN(
-    [
-      #include <pthread.h>
-      #include <string.h>
-
-      int main(int argc, char** argv) {
-        pthread_t       x1;
-        pthread_t       x2;
-        pthread_t       x3;
-
-	memset(&x1, 0x0, sizeof(x1));
-	memset(&x2, 0x0, sizeof(x2));
-	memset(&x3, 0x0, sizeof(x3));
-
-        __sync_bool_compare_and_swap(&x1, x2, x3);
-
-        return(0);
-      }
-    ],
-    [
-      AC_DEFINE([HAVE_ATOMIC_PTHREAD_T], [1],
-                [pthread_t can be used by GCC atomic builtins])
-      AC_MSG_RESULT(yes)
-    ],
-    [
-      AC_MSG_RESULT(no)
-    ]
-    )
 
   # Try using solaris atomics on SunOS if GCC atomics are not available
   AC_CHECK_DECLS(
