This patch is needed on uclibc.

-Khem

Index: udev-151/extras/ata_id/ata_id.c
===================================================================
--- udev-151.orig/extras/ata_id/ata_id.c	2010-06-04 14:08:41.912730501 -0700
+++ udev-151/extras/ata_id/ata_id.c	2010-06-04 14:09:31.492734527 -0700
@@ -168,7 +168,11 @@ static void disk_identify_fixup_uint16 (
 	uint16_t *p;
 
 	p = (uint16_t *) identify;
-	p[offset_words] = le16toh (p[offset_words]);
+#if __BYTE_ORDER == __LITTLE_ENDIAN
+	p[offset_words] = p[offset_words];
+#else
+	p[offset_words] = __bswap_16 (p[offset_words]);
+#endif
 }
 
 /**
