From 61b8233c582bcb1786bfcafa984abfd523fedcca Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Thu, 1 Dec 2011 22:07:38 +0100
Subject: [PATCH] python-elementary: temporary fix for build breakage

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 .../include/elementary/c_elementary.pxd            |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/elementary/c_elementary.pxd b/include/elementary/c_elementary.pxd
index 16535c5..09986f8 100644
--- a/include/elementary/c_elementary.pxd
+++ b/include/elementary/c_elementary.pxd
@@ -169,13 +169,13 @@ cdef extern from "Elementary.h":
         evas.c_evas.Eina_Bool hover_top
         evas.c_evas.Eina_Bool hover_bottom
 
-    ctypedef char *(*GenlistItemLabelGetFunc)(void *data, evas.c_evas.Evas_Object *obj, const_char_ptr part)
+    ctypedef char *(*GenlistItemTextGetFunc)(void *data, evas.c_evas.Evas_Object *obj, const_char_ptr part)
     ctypedef evas.c_evas.Evas_Object *(*GenlistItemIconGetFunc)(void *data, evas.c_evas.Evas_Object *obj, const_char_ptr part)
     ctypedef evas.c_evas.Eina_Bool (*GenlistItemStateGetFunc)(void *data, evas.c_evas.Evas_Object *obj, const_char_ptr part)
     ctypedef void (*GenlistItemDelFunc)(void *data, evas.c_evas.Evas_Object *obj)
 
     ctypedef struct Elm_Genlist_Item_Class_Func:
-        GenlistItemLabelGetFunc label_get
+        GenlistItemTextGetFunc text_get
         GenlistItemIconGetFunc content_get
         GenlistItemStateGetFunc state_get
         GenlistItemDelFunc del_ "del"
@@ -184,13 +184,13 @@ cdef extern from "Elementary.h":
         char *item_style
         Elm_Genlist_Item_Class_Func func
 
-    ctypedef char *(*GengridItemLabelGetFunc)(void *data, evas.c_evas.Evas_Object *obj, const_char_ptr part)
+    ctypedef char *(*GengridItemTextGetFunc)(void *data, evas.c_evas.Evas_Object *obj, const_char_ptr part)
     ctypedef evas.c_evas.Evas_Object *(*GengridItemIconGetFunc)(void *data, evas.c_evas.Evas_Object *obj, const_char_ptr part)
     ctypedef evas.c_evas.Eina_Bool (*GengridItemStateGetFunc)(void *data, evas.c_evas.Evas_Object *obj, const_char_ptr part)
     ctypedef void (*GengridItemDelFunc)(void *data, evas.c_evas.Evas_Object *obj)
 
     ctypedef struct Elm_Gengrid_Item_Class_Func:
-        GengridItemLabelGetFunc label_get
+        GengridItemTextGetFunc text_get
         GengridItemIconGetFunc content_get
         GengridItemStateGetFunc state_get
         GengridItemDelFunc del_ "del"
-- 
1.7.8.rc4

