--- a/ubi-utils/libiniparser.c	2012-05-29 23:46:08.000000000 +0200
+++ b/ubi-utils/libiniparser.c	2012-05-29 23:45:01.000000000 +0200
@@ -327,6 +327,8 @@
   the notfound value is returned.
  */
 /*--------------------------------------------------------------------------*/
+/* Floating-point is not supported in klibc */
+/*
 double iniparser_getdouble(dictionary * d, char * key, double notfound)
 {
     char    *   str ;
@@ -335,6 +337,7 @@
     if (str==INI_INVALID_KEY) return notfound ;
     return atof(str);
 }
+*/
 
 /*-------------------------------------------------------------------------*/
 /**
