--- a/ubi-utils/libiniparser.c	2012-04-30 23:59:40.000000000 +0200
+++ b/ubi-utils/libiniparser.c	2012-04-30 23:55:15.000000000 +0200
@@ -333,7 +333,9 @@
 
     str = iniparser_getstring(d, key, INI_INVALID_KEY);
     if (str==INI_INVALID_KEY) return notfound ;
-    return atof(str);
+    double value;
+    sscanf(str,"%lf",&value);
+    return value;
 }
 
 /*-------------------------------------------------------------------------*/
