--- a/ubi-utils/libubi.c	2012-05-01 01:57:51.000000000 +0200
+++ b/ubi-utils/libubi.c	2012-05-01 01:55:14.000000000 +0200
@@ -686,7 +706,7 @@
  * success and %-1 in case of failure. @r->ubi_num contains newly created UBI
  * device number.
  */
-static int do_attach(const char *node, const struct ubi_attach_req *r)
+static int do_attach(const char *node, struct ubi_attach_req *r)
 {
 	int fd, ret;
 
