Upstream-Status: Submitted
https://code.google.com/p/omgps/issues/detail?id=15

--- a/src/settings.c	2011-12-12 13:21:30.573400795 +0100
+++ b/src/settings.c	2011-12-12 14:11:05.269606119 +0100
@@ -278,6 +278,9 @@
 	}
 
 	ret = check_settings(errbuf, ERRBUF_LEN);
+	if (ret == FALSE) {
+		snprintf(errbuf, ERRBUF_LEN, "Something wrong while loading settings file");
+	}
 
 	if (fd > 0)
 		close(fd);
