Upstream-Status: Backport

from bzr changeset 99307

=== modified file 'ChangeLog'
--- ChangeLog	2010-01-12 05:11:05 +0000
+++ ChangeLog	2010-01-12 16:25:08 +0000
@@ -1,3 +1,7 @@
+2010-01-12  Chong Yidong  <cyd@stupidchicken.com>
+
+	* configure.in: Explicitly check for and link to -lXrender.
+
 2010-01-12  Glenn Morris  <rgm@gnu.org>
 
 	* INSTALL.BZR, README: Use bug-gnu-emacs rather than emacs-pretest-bug

=== modified file 'configure.in'
--- configure.in	2010-01-04 05:35:18 +0000
+++ configure.in	2010-01-12 16:25:08 +0000
@@ -1984,12 +1984,17 @@
     if test "x${with_xft}" != "xno"; then
 
       PKG_CHECK_MODULES(XFT, xft >= 0.13.0, , HAVE_XFT=no)
-      if test "$HAVE_XFT" != no; then
+      ## Because xftfont.c uses XRenderQueryExtension, we also
+      ## need to link to -lXrender.
+      HAVE_XRENDER=no
+      AC_CHECK_LIB(Xrender, XRenderQueryExtension, HAVE_XRENDER=yes)
+      if test "$HAVE_XFT" != no && test "$HAVE_XRENDER" != no; then
 	OLD_CPPFLAGS="$CPPFLAGS"
 	OLD_CFLAGS="$CFLAGS"
 	OLD_LIBS="$LIBS"
 	CPPFLAGS="$CPPFLAGS $XFT_CFLAGS"
 	CFLAGS="$CFLAGS $XFT_CFLAGS"
+	XFT_LIBS="-lXrender $XFT_LIBS"
 	LIBS="$XFT_LIBS $LIBS"
 	AC_CHECK_HEADER(X11/Xft/Xft.h,
 	  AC_CHECK_LIB(Xft, XftFontOpen, HAVE_XFT=yes, , $XFT_LIBS))

