Index: emacs-23.1/leim/Makefile.in
===================================================================
--- emacs-23.1.orig/leim/Makefile.in
+++ emacs-23.1/leim/Makefile.in
@@ -51,7 +51,7 @@ buildlisppath=${srcdir}/${dot}${dot}/lis
 
 # How to run Emacs.
 RUN-EMACS = EMACSLOADPATH=$(buildlisppath) LC_ALL=C \
-	${BUILT-EMACS} -batch --no-init-file --no-site-file --multibyte
+	${QEMU} ${BUILT-EMACS} -batch --no-init-file --no-site-file --multibyte
 
 # Subdirectories to be made if ${srcdir} is different from the current
 # directory.
Index: emacs-23.1/lib-src/Makefile.in
===================================================================
--- emacs-23.1.orig/lib-src/Makefile.in
+++ emacs-23.1/lib-src/Makefile.in
@@ -23,7 +23,7 @@
 SHELL = /bin/sh
 
 # Following ../lisp/Makefile.in.
-EMACS = ../src/emacs
+EMACS = ${QEMU} ../src/emacs
 EMACSOPT = -batch --no-site-file --multibyte
 
 # ==================== Things `configure' will edit ====================
@@ -371,7 +371,7 @@ TAGS: etags${EXEEXT}
    clobbered too.  */
 test-distrib${EXEEXT}: ${srcdir}/test-distrib.c
 	$(CC) ${ALL_CFLAGS} -o test-distrib ${srcdir}/test-distrib.c
-	./test-distrib ${srcdir}/testfile
+	${QEMU} ./test-distrib ${srcdir}/testfile
 
 /* We need the following in order to create a <getopt.h> when the system
    does not have one that works with the given compiler.  */
Index: emacs-23.1/lisp/Makefile.in
===================================================================
--- emacs-23.1.orig/lisp/Makefile.in
+++ emacs-23.1/lisp/Makefile.in
@@ -26,8 +26,7 @@ srcdir=@srcdir@/..
 # You can specify a different executable on the make command line,
 # e.g. "make EMACS=../src/emacs ...".
 
-EMACS = ../src/emacs
-
+EMACS = "${QEMU} ../src/emacs"
 # Command line flags for Emacs.  This must include --multibyte,
 # otherwise some files will not compile.
 
Index: emacs-23.1/Makefile.in
===================================================================
--- emacs-23.1.orig/Makefile.in
+++ emacs-23.1/Makefile.in
@@ -336,7 +336,7 @@ lib-src lisp: Makefile FRC
 # all preloaded elisp files, and only then dump the actual src/emacs, which
 # is not wrong, but is overkill in 99.99% of the cases.
 src: Makefile FRC
-	boot=bootstrap-emacs$(EXEEXT);                         \
+	boot=${QEMU} bootstrap-emacs$(EXEEXT);                         \
 	if [ ! -x "src/$$boot" ]; then                                     \
 	    cd $@; $(MAKE) all $(MFLAGS)                                   \
 	      CC='${CC}' CFLAGS='${CFLAGS}' CPPFLAGS='${CPPFLAGS}'         \
Index: emacs-23.1/src/Makefile.in
===================================================================
--- emacs-23.1.orig/src/Makefile.in
+++ emacs-23.1/src/Makefile.in
@@ -497,7 +497,7 @@
    this with the shell''s ``for'' construct.
    Note that some people do not have '.'  in their paths, so we must
    use ./prefix-args.  */
-#define YMF_PASS_LDFLAGS(flags) `./prefix-args -Xlinker flags`
+#define YMF_PASS_LDFLAGS(flags) `${QEMU} ./prefix-args -Xlinker flags`
 #else
 #define YMF_PASS_LDFLAGS(flags) flags
 #endif
@@ -895,9 +895,9 @@ LIBES = $(LOADLIBES) $(LIBS) $(LIBX) $(L
 #endif
 
 #ifdef HAVE_SHM
-RUN_TEMACS = `/bin/pwd`/temacs -nl
+RUN_TEMACS = ${QEMU} temacs -nl
 #else
-RUN_TEMACS = `/bin/pwd`/temacs
+RUN_TEMACS = ${QEMU} temacs
 #endif
 
 all: emacs${EXEEXT} OTHER_FILES
@@ -912,7 +912,7 @@ emacs${EXEEXT}: temacs${EXEEXT} ${etc}DO
 	@: bootstrap-emacs, so let us replace it.
 	-ln -f emacs${EXEEXT} bootstrap-emacs${EXEEXT}
 #endif /* ! defined (CANNOT_DUMP) */
-	-./emacs -q -batch -f list-load-path-shadows
+	-${QEMU} ./emacs -q -batch -f list-load-path-shadows
 
 /* We run make-docfile twice because the command line may get too long
    on some systems.  */
@@ -929,8 +929,8 @@ emacs${EXEEXT}: temacs${EXEEXT} ${etc}DO
    only in order to reduce the command line length.  --Stef  */
 ${etc}DOC: ${libsrc}make-docfile${EXEEXT} ${obj} ${lisp} ${SOME_MACHINE_LISP}
 	-rm -f ${etc}DOC
-	${libsrc}make-docfile -d ${srcdir} ${SOME_MACHINE_OBJECTS} ${obj} > ${etc}DOC
-	${libsrc}make-docfile -a ${etc}DOC -d ${srcdir} ${SOME_MACHINE_LISP} ${shortlisp}
+	${QEMU}${libsrc}make-docfile -d ${srcdir} ${SOME_MACHINE_OBJECTS} ${obj} > ${etc}DOC
+	${QEMU} ${libsrc}make-docfile -a ${etc}DOC -d ${srcdir} ${SOME_MACHINE_LISP} ${shortlisp}
 
 ${libsrc}make-docfile${EXEEXT}:
 	cd ${libsrc}; ${MAKE} ${MFLAGS} make-docfile${EXEEXT}
