Index: SDL_mixer-1.2.11/configure.in
===================================================================
--- SDL_mixer-1.2.11.orig/configure.in	2011-02-15 11:54:23.929086120 +0100
+++ SDL_mixer-1.2.11/configure.in	2011-02-15 11:56:53.745736645 +0100
@@ -1,6 +1,5 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_INIT(README)
-AC_CONFIG_AUX_DIRS($srcdir/build-scripts)
 
 dnl Set various version strings - taken gratefully from the GTk sources
 
@@ -40,10 +39,6 @@
 AC_SUBST(LT_REVISION)
 AC_SUBST(LT_AGE)
 
-dnl Detect the canonical build and host environments
-AC_CONFIG_AUX_DIRS($srcdir/build-scripts)
-dnl AC_CANONICAL_HOST
-
 dnl Check for tools
 AC_PROG_LIBTOOL
 AC_PROG_CC
