--- /tmp/configure.in	2009-12-02 16:18:33.000000000 +0100
+++ SDL_mixer-1.2.11/configure.in	2009-12-02 16:20:02.000000000 +0100
@@ -413,24 +413,7 @@
               [], [enable_music_flac_shared=yes])
 if test x$enable_music_flac = xyes; then
     AC_CHECK_HEADER([FLAC/export.h], [have_flac_export=yes])
-    if test x$have_flac_export = xyes; then
-        LIBS_SAVED="$LIBS"
-        LIBS="-lFLAC"
-        AC_MSG_CHECKING([for libflac so-name version >= $libflac_ver])
-        AC_TRY_RUN([
-#include "FLAC/export.h"
-#include "stdio.h"
-int main( int argc, char *argv[] ) {
-#if defined(FLAC_API_VERSION_CURRENT) && (FLAC_API_VERSION_CURRENT >= $libflac_ver)
-    return 0;
-#else
-    return 1;
-#endif
-}
-], have_flac_ver=yes, have_flac_ver=no)
-        LIBS="$LIBS_SAVED"
-        AC_MSG_RESULT($have_flac_ver)
-    fi
+	have_flac_ver=yes
 
     if test x$have_flac_ver = xyes; then
         AC_CHECK_HEADER([FLAC/stream_decoder.h], [have_flac_hdr=yes])
