Index: git/src/nspawn/nspawn.c
===================================================================
--- git.orig/src/nspawn/nspawn.c	2012-09-22 17:57:05.000000000 -0700
+++ git/src/nspawn/nspawn.c	2012-09-22 18:01:41.541405768 -0700
@@ -63,6 +63,8 @@
         LINK_GUEST
 } LinkJournal;
 
+#include "config.h"
+
 static char *arg_directory = NULL;
 static char *arg_user = NULL;
 static char **arg_controllers = NULL;
@@ -1373,7 +1375,12 @@
                                 a[0] = (char*) "/sbin/init";
                                 execve(a[0], a, (char**) envp);
                         } else if (argc > optind)
+#ifdef HAVE_EXECVPE
                                 execvpe(argv[optind], argv + optind, (char**) envp);
+#else
+                                environ = (char **)envp;
+                                execvp(argv[optind], argv + optind);
+#endif /* HAVE_EXECVPE */			
                         else {
                                 chdir(home ? home : "/root");
                                 execle("/bin/bash", "-bash", NULL, (char**) envp);
