From f5560d9557ee48fb79810180ddfd3ec386e2a7b5 Mon Sep 17 00:00:00 2001
From: Milo Yip <miloyip@gmail.com>
Date: Wed, 2 Mar 2016 01:01:17 +0800
Subject: [PATCH] Fix gcc strict-overflow warning

Fix #566 #568

Upstream-Status: Backport [Partial merge of upstream commit 928caf92e]

Signed-off-by: Andre McCurdy <armccurdy@gmail.com>
---
 include/rapidjson/internal/dtoa.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/rapidjson/internal/dtoa.h b/include/rapidjson/internal/dtoa.h
index 2d8d2e4..15571e1 100644
--- a/include/rapidjson/internal/dtoa.h
+++ b/include/rapidjson/internal/dtoa.h
@@ -148,7 +148,7 @@ inline char* WriteExponent(int K, char* buffer) {
 inline char* Prettify(char* buffer, int length, int k) {
     const int kk = length + k;  // 10^(kk-1) <= v < 10^kk
 
-    if (length <= kk && kk <= 21) {
+    if (0 <= k && kk <= 21) {
         // 1234e7 -> 12340000000
         for (int i = length; i < kk; i++)
             buffer[i] = '0';
-- 
1.9.1

