--- /tmp/system_info.py	2008-08-22 00:38:16.000000000 +0200
+++ numpy-1.1.1/numpy/distutils/system_info.py	2008-08-22 00:40:33.013198000 +0200
@@ -137,11 +137,8 @@
     default_x11_lib_dirs = []
     default_x11_include_dirs = []
 else:
-    default_lib_dirs = ['/usr/local/lib', '/opt/lib', '/usr/lib',
-                        '/opt/local/lib', '/sw/lib']
-    default_include_dirs = ['/usr/local/include',
-                            '/opt/include', '/usr/include',
-                            '/opt/local/include', '/sw/include']
+    default_lib_dirs = ['/non-existant-dir']
+    default_include_dirs = ['non-existant-dir']
     default_src_dirs = ['.','/usr/local/src', '/opt/src','/sw/src']
 
     try:
