From 5b22f64a2e2237082d2733698b07147d27b09ad2 Mon Sep 17 00:00:00 2001
From: Michael Hudson-Doyle <michael.hudson@linaro.org>
Date: Fri, 20 Dec 2013 12:28:27 +1300
Subject: [PATCH 2/5] Fix linking when scripting is disabled.

---
 src/mongo/scripting/engine_none.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/mongo/scripting/engine_none.cpp b/src/mongo/scripting/engine_none.cpp
index f5c7109..9ae9d57 100644
--- a/src/mongo/scripting/engine_none.cpp
+++ b/src/mongo/scripting/engine_none.cpp
@@ -33,4 +33,8 @@ namespace mongo {
     void ScriptEngine::setup() {
         // noop
     }
+
+    std::string ScriptEngine::getInterpreterVersionString() {
+        return "none";
+    }
 }
-- 
1.9.0

