diff -uNr dfu-util-0.4.orig/configure.ac dfu-util-0.4/configure.ac
--- dfu-util-0.4.orig/configure.ac	2014-07-18 08:18:45.407552416 +0200
+++ dfu-util-0.4/configure.ac	2014-07-18 20:05:11.934894143 +0200
@@ -4,7 +4,7 @@
 AC_PREREQ(2.59)
 AC_INIT([dfu-util],[0.4])
 AC_CONFIG_AUX_DIR(m4)
-AM_INIT_AUTOMAKE(AC_PACKAGE_NAME, AC_PACKAGE_VERSION)
+AM_INIT_AUTOMAKE([foreign])
 AM_CONFIG_HEADER([config.h])
 
 # Test for new silent rules and enable only if they are available
