Upstream-Status: Pending

Index: gcc-4_6-branch/Makefile.def
===================================================================
--- gcc-4_6-branch.orig/Makefile.def	2012-03-03 01:08:03.000000000 -0800
+++ gcc-4_6-branch/Makefile.def	2012-03-04 09:19:02.430607447 -0800
@@ -242,6 +242,7 @@
 flags_to_pass = { flag= BISON ; };
 flags_to_pass = { flag= CC_FOR_BUILD ; };
 flags_to_pass = { flag= CFLAGS_FOR_BUILD ; };
+flags_to_pass = { flag= CPPFLAGS_FOR_BUILD ; };
 flags_to_pass = { flag= CXX_FOR_BUILD ; };
 flags_to_pass = { flag= EXPECT ; };
 flags_to_pass = { flag= FLEX ; };
Index: gcc-4_6-branch/gcc/Makefile.in
===================================================================
--- gcc-4_6-branch.orig/gcc/Makefile.in	2012-03-03 01:03:17.000000000 -0800
+++ gcc-4_6-branch/gcc/Makefile.in	2012-03-04 09:19:02.430607447 -0800
@@ -770,7 +770,7 @@
 
 # Native linker and preprocessor flags.  For x-fragment overrides.
 BUILD_LDFLAGS=@BUILD_LDFLAGS@
-BUILD_CPPFLAGS=$(ALL_CPPFLAGS)
+BUILD_CPPFLAGS=$(INCLUDES) @BUILD_CPPFLAGS@ $(X_CPPFLAGS)
 
 # Actual name to use when installing a native compiler.
 GCC_INSTALL_NAME := $(shell echo gcc|sed '$(program_transform_name)')
Index: gcc-4_6-branch/gcc/configure.ac
===================================================================
--- gcc-4_6-branch.orig/gcc/configure.ac	2012-03-03 01:17:45.000000000 -0800
+++ gcc-4_6-branch/gcc/configure.ac	2012-03-04 09:19:02.430607447 -0800
@@ -1774,16 +1774,18 @@
 # Also, we cannot run fixincludes.
 
 # These are the normal (build=host) settings:
-CC_FOR_BUILD='$(CC)'		AC_SUBST(CC_FOR_BUILD)
-BUILD_CFLAGS='$(ALL_CFLAGS)'	AC_SUBST(BUILD_CFLAGS)
-BUILD_LDFLAGS='$(LDFLAGS)'	AC_SUBST(BUILD_LDFLAGS)
-STMP_FIXINC=stmp-fixinc		AC_SUBST(STMP_FIXINC)
+CC_FOR_BUILD='$(CC)'			AC_SUBST(CC_FOR_BUILD)
+BUILD_CFLAGS='$(ALL_CFLAGS)'		AC_SUBST(BUILD_CFLAGS)
+BUILD_LDFLAGS='$(LDFLAGS)'		AC_SUBST(BUILD_LDFLAGS)
+BUILD_CPPFLAGS='$(ALL_CPPFLAGS)'	AC_SUBST(BUILD_CPPFLAGS)
+STMP_FIXINC=stmp-fixinc			AC_SUBST(STMP_FIXINC)
 
 # And these apply if build != host, or we are generating coverage data
 if test x$build != x$host || test "x$coverage_flags" != x
 then
     BUILD_CFLAGS='$(INTERNAL_CFLAGS) $(T_CFLAGS) $(CFLAGS_FOR_BUILD)'
     BUILD_LDFLAGS='$(LDFLAGS_FOR_BUILD)'
+    BUILD_CPPFLAGS='$(CPPFLAGS_FOR_BUILD)'
 fi
 
 # Expand extra_headers to include complete path.
Index: gcc-4_6-branch/Makefile.in
===================================================================
--- gcc-4_6-branch.orig/Makefile.in	2012-03-03 01:08:03.000000000 -0800
+++ gcc-4_6-branch/Makefile.in	2012-03-04 09:19:02.446607448 -0800
@@ -338,6 +338,7 @@
 AS_FOR_BUILD = @AS_FOR_BUILD@
 CC_FOR_BUILD = @CC_FOR_BUILD@
 CFLAGS_FOR_BUILD = @CFLAGS_FOR_BUILD@
+CPPFLAGS_FOR_BUILD = @CPPFLAGS_FOR_BUILD@
 CXXFLAGS_FOR_BUILD = @CXXFLAGS_FOR_BUILD@
 CXX_FOR_BUILD = @CXX_FOR_BUILD@
 DLLTOOL_FOR_BUILD = @DLLTOOL_FOR_BUILD@
@@ -691,6 +692,7 @@
 	"BISON=$(BISON)" \
 	"CC_FOR_BUILD=$(CC_FOR_BUILD)" \
 	"CFLAGS_FOR_BUILD=$(CFLAGS_FOR_BUILD)" \
+	"CPPFLAGS_FOR_BUILD=$(CPPFLAGS_FOR_BUILD)" \
 	"CXX_FOR_BUILD=$(CXX_FOR_BUILD)" \
 	"EXPECT=$(EXPECT)" \
 	"FLEX=$(FLEX)" \
Index: gcc-4_6-branch/gcc/configure
===================================================================
--- gcc-4_6-branch.orig/gcc/configure	2012-03-03 01:17:45.000000000 -0800
+++ gcc-4_6-branch/gcc/configure	2012-03-04 09:19:15.638608087 -0800
@@ -703,6 +703,7 @@
 LIBTOOL
 collect2
 STMP_FIXINC
+BUILD_CPPFLAGS
 BUILD_LDFLAGS
 BUILD_CFLAGS
 CC_FOR_BUILD
@@ -4842,7 +4843,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $acx_cv_cc_gcc_supports_ada" >&5
 $as_echo "$acx_cv_cc_gcc_supports_ada" >&6; }
 
-if test x$GNATBIND != xno && test x$GNATMAKE != xno && test x$acx_cv_cc_gcc_supports_ada != xno; then
+if test "x$GNATBIND" != xno && test "x$GNATMAKE" != xno && test x$acx_cv_cc_gcc_supports_ada != xno; then
   have_gnat=yes
 else
   have_gnat=no
@@ -11372,6 +11373,7 @@
 CC_FOR_BUILD='$(CC)'
 BUILD_CFLAGS='$(ALL_CFLAGS)'
 BUILD_LDFLAGS='$(LDFLAGS)'
+BUILD_CPPFLAGS='$(ALL_CPPFLAGS)'
 STMP_FIXINC=stmp-fixinc
 
 # And these apply if build != host, or we are generating coverage data
@@ -11379,6 +11381,7 @@
 then
     BUILD_CFLAGS='$(INTERNAL_CFLAGS) $(T_CFLAGS) $(CFLAGS_FOR_BUILD)'
     BUILD_LDFLAGS='$(LDFLAGS_FOR_BUILD)'
+    BUILD_CPPFLAGS='$(CPPFLAGS_FOR_BUILD)'
 fi
 
 # Expand extra_headers to include complete path.
@@ -17495,7 +17498,7 @@
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<_LT_EOF
-#line 17498 "configure"
+#line 17501 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -17601,7 +17604,7 @@
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<_LT_EOF
-#line 17604 "configure"
+#line 17607 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
Index: gcc-4_6-branch/Makefile.tpl
===================================================================
--- gcc-4_6-branch.orig/Makefile.tpl	2012-03-03 01:08:03.000000000 -0800
+++ gcc-4_6-branch/Makefile.tpl	2012-03-04 09:19:02.454607448 -0800
@@ -341,6 +341,7 @@
 AS_FOR_BUILD = @AS_FOR_BUILD@
 CC_FOR_BUILD = @CC_FOR_BUILD@
 CFLAGS_FOR_BUILD = @CFLAGS_FOR_BUILD@
+CPPFLAGS_FOR_BUILD = @CPPFLAGS_FOR_BUILD@
 CXXFLAGS_FOR_BUILD = @CXXFLAGS_FOR_BUILD@
 CXX_FOR_BUILD = @CXX_FOR_BUILD@
 DLLTOOL_FOR_BUILD = @DLLTOOL_FOR_BUILD@
Index: gcc-4_6-branch/configure.ac
===================================================================
--- gcc-4_6-branch.orig/configure.ac	2012-03-03 01:17:45.000000000 -0800
+++ gcc-4_6-branch/configure.ac	2012-03-04 09:19:02.454607448 -0800
@@ -3123,6 +3123,7 @@
 # our build compiler if desired.
 if test x"${build}" = x"${host}" ; then
   CFLAGS_FOR_BUILD=${CFLAGS_FOR_BUILD-${CFLAGS}}
+  CPPFLAGS_FOR_BUILD=${CPPFLAGS_FOR_BUILD-${CPPFLAGS}}
   CXXFLAGS_FOR_BUILD=${CXXFLAGS_FOR_BUILD-${CXXFLAGS}}
   LDFLAGS_FOR_BUILD=${LDFLAGS_FOR_BUILD-${LDFLAGS}}
 fi
@@ -3189,6 +3190,7 @@
 AC_SUBST(AS_FOR_BUILD)
 AC_SUBST(CC_FOR_BUILD)
 AC_SUBST(CFLAGS_FOR_BUILD)
+AC_SUBST(CPPFLAGS_FOR_BUILD)
 AC_SUBST(CXXFLAGS_FOR_BUILD)
 AC_SUBST(CXX_FOR_BUILD)
 AC_SUBST(DLLTOOL_FOR_BUILD)
Index: gcc-4_6-branch/configure
===================================================================
--- gcc-4_6-branch.orig/configure	2012-03-03 01:17:45.000000000 -0800
+++ gcc-4_6-branch/configure	2012-03-04 09:19:02.458607448 -0800
@@ -617,6 +617,7 @@
 DLLTOOL_FOR_BUILD
 CXX_FOR_BUILD
 CXXFLAGS_FOR_BUILD
+CPPFLAGS_FOR_BUILD
 CFLAGS_FOR_BUILD
 CC_FOR_BUILD
 AS_FOR_BUILD
@@ -7644,6 +7645,7 @@
 # our build compiler if desired.
 if test x"${build}" = x"${host}" ; then
   CFLAGS_FOR_BUILD=${CFLAGS_FOR_BUILD-${CFLAGS}}
+  CPPFLAGS_FOR_BUILD=${CPPFLAGS_FOR_BUILD-${CPPFLAGS}}
   CXXFLAGS_FOR_BUILD=${CXXFLAGS_FOR_BUILD-${CXXFLAGS}}
   LDFLAGS_FOR_BUILD=${LDFLAGS_FOR_BUILD-${LDFLAGS}}
 fi
@@ -7709,6 +7711,7 @@
 
 
 
+
 
 
 
