From 50a9133e0b13798640da0b7cd11b5a85e35b2038 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Fri, 23 Nov 2018 16:27:32 +0800
Subject: [PATCH] meson.build: do not compile docs

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/meson.build b/meson.build
index 04b6d47..46fdbc8 100644
--- a/meson.build
+++ b/meson.build
@@ -81,7 +81,6 @@ subdir('gdbus')
 subdir('gsettings')
 subdir('client')
 subdir('bin')
-subdir('docs')
 subdir('tests')
 
 meson.add_install_script('meson_post_install.py', gio_module_dir)
-- 
2.7.4

